---
title: "16S rRNA gene amplicon analysis - SafeVir - DIO - Termination"
output:
  pdf_document: default
  html_document:
    number_sections: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, include=F}
#Set working directory to script directory
setwd(dirname(rstudioapi::getActiveDocumentContext()$path))

#Load and prepare OTU-tables + color scale etc.
source("Analysis0_file_loading_and_prep2.R")
```

# Sequencing depth
```{r eval=T, echo=F, message=FALSE, include=T}
PSBamp <- phyloseq_to_ampvis2(PSB)

PSB.CSSamp <- phyloseq_to_ampvis2(PSB.CSS)

amp_rarecurve(
  PSBamp,
  stepsize = 1000,
  color_by = "ProjectID",
  facet_scales = "fixed"
) + scale_color_viridis(discrete=TRUE) +
  geom_dl(aes(label = ProjectID), method = list(dl.trans(x = x + 0.0), "last.points", cex = 0.8))
  
```

# Individual sample barplots.

```{r, eval = T, include = T, echo = F, message = F}

bac.phyl <- tax_glom(PSB, "Genus", NArm = FALSE)

ps0 <- transform_sample_counts(bac.phyl, function(x) x / sum(x))

#ps1<- merge_samples(ps0, "Time")

#ps1 <- transform_sample_counts(ps0, function(x) x / sum(x))


#Create melted dataframe
df <- psmelt(ps0)

#Select last non-empty taxonomic rank
df[df==""] <- NA

df$tax <- apply(df, 1, function(x) tail(na.omit(x), 1))

#Arrange samples by mean abundance
top <- df %>%
  group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  arrange(-Mean)
#Show top
#top

top10 <- top$tax[1:150]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))

df0 <- df0[order(df0$Sample, decreasing = TRUE), ]

barplot.all.samples <- ggplot(df0, aes(ProjectID, Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(width = 0.8) +
  scale_fill_jco(name = "Taxonomy") +
  scale_fill_manual(values=rep(col_fil,5),name="Genus")+
  theme_classic() +
  theme(text = element_text(size = 10, colour = "Black"),
        axis.line=element_line(size=0.5),
        #panel.border = element_blank(),
        axis.text=element_text(size = 10, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(angle = 0, size=12, face = "bold"),
        #legend.position = "none"
  ) +
  ylab("Relative abundance")


barplot.all.samples

```
## Barplots based on treatments. 

 
```{r, eval = T, include = T, echo = F, message = F}

vir.phyl <- tax_glom(PSB, "Genus", NArm = FALSE)

ps0 <- transform_sample_counts(vir.phyl, function(x) x / sum(x))

ps1 <- merge_samples(ps0, "Treatment")

ps1 <- transform_sample_counts(ps1, function(x) x / sum(x))


#Create melted dataframe
df <- psmelt(ps1)

#Select last non-empty taxonomic rank
df[df==""] <- NA

df$tax <- apply(df, 1, function(x) tail(na.omit(x), 1))

#Arrange samples by mean abundance
top <- df %>%
  group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  arrange(-Mean)
#Show top
top

top10 <- top$tax[1:50]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))

df0 <- df0[order(df0$Sample, decreasing = TRUE), ]

#Set order for samples
#df0$Treatment <- factor(df0$Treatment, levels = c("Control", "Amp", "FOP", "LM", "LM+Amp", "LM+FOP"))

df$Sample <- factor(df0$Sample, levels = c("3", "4", "5"))

barplot.genus.treatment <- ggplot(df0, aes(Sample, Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(width = 0.8) +
  scale_fill_jco(name = "Taxonomy") +
  scale_fill_manual(values=rep(col_fil,5),name="Genus")+
  theme_classic() +
  theme(text = element_text(size = 10, colour = "Black"),
        axis.line=element_line(size=0.5),
        #panel.border = element_blank(),
        axis.text=element_text(size = 10, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(angle = 0, size=12, face = "bold"),
        #legend.position = "none"
  ) +
  ylab("Relative abundance") +
  xlab("Treatment")


barplot.genus.treatment

```

```{r, eval = T, include = T, echo = F, message = F}

##################################Verrucomicrobiaceae barplots
################By species

PSB.rel <- transform_sample_counts(PSB, function(x) x / sum(x))
ps0 <- tax_glom(subset_taxa(PSB.rel, Family=="Verrucomicrobiaceae"), "Genus", NArm = FALSE)

#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_AKM_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Akkermansia") +
  theme_classic() +
  theme(
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  xlab("") +
  ylab("Relative abundance") 
bar_AKM_species
```


```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on AKM abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```

```{r, eval = T, include = T, echo = F, message = F}

##################################Erysipelotrichaceae barplots
################By species

PSB.rel <- transform_sample_counts(PSB, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Allobaculum"), "Species", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_ALL_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Allobaculum") +
  theme_classic() +
  theme(
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  xlab("") +
  ylab("Relative abundance") 
bar_ALL_species
```


```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Allobaculum abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)
```

```{r, eval = T, include = T, echo = F, message = F}

##################################Lactobacillus barplots
################By species

PSB.rel <- transform_sample_counts(PSB, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Lactobacillus"), "Species", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_LAC_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Lactobacillus") +
  theme_classic() +
  theme(
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  xlab("") +
  ylab("Relative abundance") 
bar_LAC_species
```


```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Lactobacillus abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)
```



```{r, eval = T, include = T, echo = F, message = F}
# Boxplots for Termination

bac.phyl <- tax_glom(PSB, "Genus", NArm = FALSE)
ps0 <- transform_sample_counts(bac.phyl, function(x) x / sum(x))

#Select desired rank
df <- psmelt(ps0)

#Select last non-empty taxonomic rank
df[df==""] <- NA

df$tax <- apply(df, 1, function(x) tail(na.omit(x), 1))

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  arrange(-Mean)
#Show top
top

top10 <- top$tax[1:100]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))

bar_bac_genus_boxplot <- ggplot(df0, aes(Treatment, Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_boxplot(show.legend = FALSE) +
  scale_fill_jco() +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=90, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  scale_y_continuous(breaks=c(0.00,0.25,0.5)) +
  xlab("") +
  ylab("Relative abundance") +
  facet_grid(tax~.,scales="free") 

bar_bac_genus_boxplot
```

# Abundance heatmaps
## Bacteria  - Treatment

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE}
#Load phyloseq files to ampvis2 format
PSBamp <- phyloseq_to_ampvis2(PSB)  

#Bacteriome

amp_heatmap(PSBamp,
            group_by = "Treatment",
            #facet_by = "Treatment",
            plot_values = FALSE,
            tax_show = 20,
            tax_aggregate = "Genus" ,
            #tax_add = "Family",
            tax_empty = "best",
            normalise = TRUE,
            color_vector = c("white", "red"),
            plot_colorscale = "sqrt",
            plot_legendbreaks = c(1, 10, 25, 50)
) +
  theme_classic() +
  theme(strip.background = element_rect(colour = "black", fill = "white"),
        strip.text.x = element_text(face = "bold"),
        axis.text.x=element_text(angle = -45, hjust = 0,vjust = 0.2))
```

# Alpha diversity
## By Treatment
```{r, eval = T, include = T, echo = F, message = F, warning=FALSE}

#Normalize to mean read count
standf = function(x, t=total) round(t * (x / sum(x)))
total = median(sample_sums(PSB))
PSB.R = transform_sample_counts(PSB, standf)

#Set desired alpha diversity metrics

alpha_met <- c("Observed","Shannon","InvSimpson")


###########Boxplots

variable <- c("Treatment")

alpha.Treatment <- plot_richness(PSB.R, measures=alpha_met , x=variable, color=variable) + 
  geom_boxplot(alpha=0.1, lwd = 1) +
  #stat_compare_means(comparisons = my_comparisons, method = "kruskal", paired = FALSE, symnum.args = symnum.args) +
  scale_color_jco() +
  theme_classic() +
  theme(text = element_text(size = 8),
        axis.line=element_line(size=1),
        axis.text=element_text(size = 8, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(size = 8, angle = 0),
        legend.text = element_text(size = 8)
  ) +
  labs(x="Treatment",
       y="Alpha diversity")

alpha.Treatment
```
Anova and linear regression analysis
```{r, eval = T, include = T, echo = F, warning = F}
##Richness - all time points

richness = estimate_richness(PSB.R, measures = alpha_met)

#Time

rich <- cbind(richness, variable = sample_data(PSB.R)$Treatment)

TukeyHSD(aov(Shannon ~ variable, rich))


rich$variable <- fct_relevel(rich$variable, "HFD_control") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)

rich$variable <- fct_relevel(rich$variable, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)
```

# Beta diversity
## basically the mouse ID. 
```{r, eval = T, include = T, echo = F, massage = F}
#Do PCoA ordination based on Bray-curtis distance
GP.ord <- ordinate(PSB.CSS, "PCoA", "bray")
beta.id = phyloseq::plot_ordination(PSB.CSS, GP.ord, color="Treatment") + 
  #stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  #scale_fill_manual(values= col_fil) +
  #scale_color_manual(values = col_fil) +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank()
  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
  geom_dl(aes(label = Treatment), method = list(dl.trans(x = x + 0.0), "last.points", cex = 0.8))
beta.id
```

## Treatment
```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank()
  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome") 
beta.treatment
```
Adonis and pairwise permanova analysis - Treatment
```{r, eval = T, include = T, echo = F, message = F}
library(mctoolsr)
bray.PSB <- phyloseq::distance(PSB.CSS, method = "bray")
# make a data frame from the sample_data
sampledf.PSB <- data.frame(sample_data(PSB.CSS))

adonis.PSB <- adonis2(bray.PSB ~ Treatment, method = "bray", data = sampledf.PSB, permutations = 999)
adonis.PSB

mapping_file = data.frame(sample_data(PSB.CSS))
pairwiseperm <- mctoolsr::calc_pairwise_permanovas(bray.PSB, mapping_file, "Treatment")
pairwiseperm
```


### Bacteriome - Deseq2 - Treatment - Collapsed on Genus-level - Comparing ChP vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- tax_glom(PSB, "Genus", NArm = FALSE)

PSB.pseu <- subset_samples(PSB, Treatment %in% c("FVT_ChP","HFD_control"))

otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)



diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds = DESeq2::DESeq(diagdds, test="Wald", fitType="parametric")

res = results(diagdds, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  #theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Treatment")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel))


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)

#Remove Realm from tax_table
#PSB.diff.rel@tax_table <- tax_table(PSB.diff.rel)[,c(1,3:8)]

ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic() +
  theme(axis.text.x=element_text(angle = 45, hjust = 0.9,vjust = 0.95)) +
  ggtitle("Differentially abundant bacterial species")
```


### Bacteriome - Deseq2 - Treatment - Collapsed on Genus-level

```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- tax_glom(PSB, "Genus", NArm = FALSE)

PSB.pseu <- PSB.pseu

#PSB.pseu@sam_data$Days <- as.factor(PSB.pseu@sam_data$Days)

otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)



diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds = DESeq2::DESeq(diagdds, test="Wald", fitType="parametric")

res = results(diagdds, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  #theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Treatment")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.rel, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel))


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)

#Remove Realm from tax_table
#PSB.diff.rel@tax_table <- tax_table(PSB.diff.rel)[,c(1,3:8)]

ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic() +
  theme(axis.text.x=element_text(angle = 45, hjust = 0.9,vjust = 0.95)) +
  ggtitle("Differentially abundant bacterial species")
```

### Bacteriome - Deseq2 - Treatment

```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- PSB

#PSB.pseu@sam_data$Days <- as.factor(PSB.pseu@sam_data$Days)

otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds = DESeq2::DESeq(diagdds, test="Wald", fitType="parametric")

res = results(diagdds, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  #theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Treatment")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel))


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)

#Remove Realm from tax_table
#PSB.diff.rel@tax_table <- tax_table(PSB.diff.rel)[,c(1,3:8)]

ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Species",
            tax_show = 15,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic() +
  theme(axis.text.x=element_text(angle = 45, hjust = 0.9,vjust = 0.95)) +
  ggtitle("Differentially abundant bacterial species")
```

Deseq2 - defined comparison
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- subset_samples(PSB, Treatment %in% c("FVT_ChP","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Species",
            tax_show = 15,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic()

```





## Bacteriome - Effect-size

```{r, eval = T, include = F, echo = F, message = F}

#################################Importing-Data

#standf = function(x, t=total) round(t * (x / sum(x)))
#total = median(sample_sums(PSB))
#PSB.R = transform_sample_counts(PSB, standf)

import.otu.table <- as.data.frame(otu_table(PSB.CSS)) %>%
  #dplyr::select(-taxonomy) %>%
  mutate_all(as.numeric) #Convert to numeric where possible



cure.data = meta.bac

#import.taxonomy = read.table('Bacteriome_taxonomy_LCA_known.txt', header = T)

rownames <- rownames(cure.data)

#Remove NA colomns

cure.data <- cure.data[ , ! apply( cure.data , 2 , function(x) all(is.na(x)) ) ]

#Set numeric column class
cure.data <- cure.data %>% mutate_if(is.character,as.factor)
#Add back rownames

rownames(cure.data) <- rownames

#cure.data <- cure.data %>% mutate_if(is.character, as.numeric)

X8.nonrarefied = t(import.otu.table)                                                                  ###%%% non-rarefied


###rarefying|subsampling reads in every sample
barplot(colSums(import.otu.table))
min(colSums(import.otu.table))

rare8 <- t(import.otu.table)
#rare8 = rarefy(t(import.otu.table), sample = 450000)
X8.rarefied = as.data.frame(rare8)                                                                 ###%%% rarefied

###Subset metadata to match remaining samples after rarefaction

Y.all = cure.data[rownames(X8.rarefied),]

###Log tranformation
X = log10(X8.rarefied+1)

#Select relevant numeric column

#Y <- dplyr::select(Y.all,c("Diet", "Treatment", "AUC1_week18", "AUC2_week18", "ewat_mg", "Weight_gain_perc" , "Central_memory_T_cells_CD4_Fat", "Effector_memory_T_cells_CD4_Fat", "Naive_T_cells_CD4_Fat", "Central_memory_T_cells_CD8_Fat", "Effector_memory_T_cells_CD8_Fat", "Naive_T_cells_CD8_Fat", "CD8a_TCRab_Fat","B_cells_Fat", "Dendritic_cells_Fat","Macrophages_Fat", "M1_macrophages_Fat", "GM.CSF", "IFN.g", "IL.10", "IL.15", "IL.17_A_F", "IL.6", "KC_GRO", "MIP.2","TNF.a", "IL.22"))

Y <- dplyr::select(Y.all,c("Treatment","ewat_mg", "Weight_gain_perc","AUC1_week18", "AUC2_week18","Central_memory_T_cells_CD8_Fat", "Th_cells_Fat","M1_macrophages_Fat","B_cells_Fat","Dendritic_cells_Fat", "MIP.2","TNF.a","IL.15","IL.10"))

#Y <- dplyr::select(Y.all,c("Diet", "Treatment", "AUC1_week18", "AUC2_week18", "ewat_mg", "Weight_gain_perc" , "Central_memory_T_cells_CD4_Fat", "Effector_memory_T_cells_CD4_Fat", "Naive_T_cells_CD4_Fat", "Central_memory_T_cells_CD8_Fat", "Effector_memory_T_cells_CD8_Fat", "Naive_T_cells_CD8_Fat", "CD8a_TCRab_Fat","B_cells_Fat", "Dendritic_cells_Fat","Macrophages_Fat", "M1_macrophages_Fat", "GM.CSF", "IFN.g", "IL.10", "IL.15", "IL.17_A_F", "IL.6", "KC_GRO", "MIP.2","TNF.a", "IL.22"))

Y <- unfactor(Y)
Y <- dplyr::mutate_at(Y,1:2,funs(factor)) 

#Y <- dplyr::mutate_if(Y,is.factor, as.numeric)
#Y <- dplyr::mutate_at(Y,1:5,funs(factor)) 
#Y <- dplyr::select(Y.all,c("Bact_group", "FVT_group", "Treatment", "FVT_Treatment", "Gender"))


colnames(Y) #Check remaining columns

###Select only numeric metadata culumns

#Y  <- Y %>% mutate_all(as.numeric)

##vifcor
#collinearity <- vifcor(Y, th  =0.5) #at 0.5 threshold variables have a VIF below 10.00 (<0.21 correlation!)
#collinearity

#Select relevant columns for that are non-collinear

Y.ad = Y
#Y.ad <- dplyr::select(Y.all,c("Batch", "Time")

#Y.ad <- Y.all[8:23]

#Remove NA rows

Y.ad <- Y.ad %>% drop_na()

X.ad <- X[rownames(Y.ad),]

is.na(Y.ad)
```

## Non-constrained

#### Capscale - independent effect sizes

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE, fig.align="center"}
###############Anova on capscale - all variables

anova.m <- matrix(nrow = 0, ncol =6) #Create matrix for results
colnames(anova.m) <- c("Df","SumOfSqs","F","Pr.F","Residual.Df","Residual.SumOfSqs")                
for (i in colnames(Y)){
  anova <- as.matrix(anova(capscale(X ~ Y[,i], dist = "bray")))
  anova <- c(anova[1,],anova[2,1:2])
  anova.m <-rbind(anova.m,anova)
}
rownames(anova.m) <- colnames(Y)

#Prepare for plotting
anova.dat <- anova.m %>%
  as.data.frame %>%
  tibble::rownames_to_column("Variable") %>%
  arrange(F)

#Add adjusted p-values
anova.dat$Pr.F.adj <- p.adjust(anova.dat$Pr.F, method = 'fdr', n = ncol (Y.ad))

#Show table
#knitr::kable(anova.dat)

#Divide F by 10
anova.dat$F <- anova.dat$F / 10

##Fancy plot

anova.melt <- tidyr::pivot_longer(anova.dat,cols=c("F"), names_to='measure', values_to="value") %>%
  add_significance(p.col = "Pr.F.adj", output.col = "PrF.sym",
                   cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1),
                   symbols = c("***", "**", "*", "'", "")
                   )



fancy.anova.bar <- ggbarplot(anova.melt,
                                         x="Variable",
                                         y="value",
                                         #color = "Variable",
                                         orientation = c("horizontal"),
                                         color = NA,
                                         fill = "measure",
                                         position=position_dodge(0.7),
                                         label = anova.melt$PrF.sym,
                                         lab.size = 9,
                                         lab.vjust = 0.7,
                                         lab.hjust = -0.3
                                         
) +
  #stat_pvalue_manual(
  #  adonis.melt,  label = "PrF.sym", tip.length = 0.01
  #) +
  #geom_hline(yintercept = 0.05) +
  ggtitle("Capscale dbRDA effect size: all variables") +
  ylab(expression(R^2~"value")) +
  scale_fill_jco()

fancy.anova.bar

```

#### ADONIS - decomposed to show individual contributions of factors

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE, fig.align="center"}
##All variables

adonis <- adonis(X ~ ., data = Y, method = "bray") #All variables

#Add adjusted p-values
adonis$aov.tab$`Pr(>F).adj` <- p.adjust(adonis$aov.tab$`Pr(>F)`, method = 'fdr', n = ncol (Y.ad))

#Show table
#knitr::kable(anova.dat)

#Plotting ordered R2
adonis.dat <- as.data.frame(adonis$aov.tab) %>%
  tibble::rownames_to_column("Variable")  %>%
  filter(!Variable %in% c("Residuals","Total")) %>%
  arrange(R2)

adonis.melt <- tidyr::pivot_longer(adonis.dat,cols=c("R2"), names_to='measure', values_to="value") %>%
  add_significance(p.col = "Pr(>F).adj", output.col = "PrF.sym",
                   cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1),
                   symbols = c("***", "**", "*", "'", "")
                   )



fancy.adonis.bar <- ggbarplot(adonis.melt,
                                         x="Variable",
                                         y="value",
                                         #color = "Variable",
                                         orientation = c("horizontal"),
                                         color = NA,
                                         fill = "measure",
                                         position=position_dodge(0.7),
                                         label = adonis.melt$PrF.sym,
                                         lab.size = 9,
                                         lab.vjust = 0.7,
                                         lab.hjust = -0.3
                                         
) +
  #stat_pvalue_manual(
  #  adonis.melt,  label = "PrF.sym", tip.length = 0.01
  #) +
  #geom_hline(yintercept = 0.05) +
  ggtitle("Adonis effect size: non-collinear variables") +
  ylab(expression(R^2~"value")) +
  scale_fill_jco()

fancy.adonis.bar
```



```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```



