---
title: "16S rRNA gene amplicon analysis - SafeVir - DIO - Termination"
output:
  pdf_document: default
  html_document:
    number_sections: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, include=F}
#Set working directory to script directory
setwd(dirname(rstudioapi::getActiveDocumentContext()$path))

#Load and prepare OTU-tables + color scale etc.
# source("Analysis0_file_loading_and_prep2.R")
source("Analysis0_file_loading_and_prep.R")
```

# Sequencing depth_rarecurve
```{r eval=T, echo=F, message=FALSE, include=T}
# PSBamp <- phyloseq_to_ampvis2(PSB)
# 
# PSB.CSSamp <- phyloseq_to_ampvis2(PSB.CSS)
# 
# amp_rarecurve(
#   PSBamp,
#   stepsize = 1000,
#   color_by = "ProjectID",
#   facet_scales = "fixed"
# ) + scale_color_viridis(discrete=TRUE) +
#   geom_dl(aes(label = ProjectID), method = list(dl.trans(x = x + 0.0), "last.points", cex = 0.8))
  
```
# barplots
## Individual sample barplots.

```{r, eval = T, include = T, echo = F, message = F}

bac.phyl <- tax_glom(PSB, "Genus", NArm = FALSE)

ps0 <- transform_sample_counts(bac.phyl, function(x) x / sum(x))

#Create melted dataframe
df <- psmelt(ps0)

#Select last non-empty taxonomic rank
df[df==""] <- NA

df$tax <- apply(df, 1, function(x) tail(na.omit(x), 1))

#Arrange samples by mean abundance
top <- df %>%
  group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  arrange(-Mean)
#Show top
#top

top10 <- top$tax[1:150]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))

df0 <- df0[order(df0$Sample, decreasing = TRUE), ]

barplot.all.samples <- ggplot(df0, aes(ProjectID, Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(width = 0.8) +
  scale_fill_jco(name = "Taxonomy") +
  scale_fill_manual(values=rep(col_fil,5),name="Genus")+
  theme_classic() +
  theme(text = element_text(size = 10, colour = "Black"),
        axis.line=element_line(size=0.5),
        #panel.border = element_blank(),
        axis.text=element_text(size = 10, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(angle = 0, size=12, face = "bold"),
        #legend.position = "none"
  ) +
  ylab("Relative abundance")


barplot.all.samples

```



## Abundance bar plot - Phylum _ termination.
```{r}


PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
 vir.phyl <- tax_glom(PSB_Termination, "Phylum", NArm = FALSE)

ps0 <- transform_sample_counts(vir.phyl, function(x) x *100/ sum(x))

ps1 <- merge_samples(ps0, "Treatment")

ps1 <- transform_sample_counts(ps1, function(x) x / sum(x))


#Create melted dataframe
df <- psmelt(ps1)

#Select last non-empty taxonomic rank
df[df==""] <- NA

df$tax <- apply(df, 1, function(x) tail(na.omit(x), 1))

#Arrange samples by mean abundance
# top <- df %>%
#   group_by(Sample, tax) %>%
#   dplyr::summarize(Mean = mean(Abundance)) %>%
#   arrange(-Mean)

top<-df %>%
  # filter(level=="Phylum")%>%
  group_by(Treatment,ProjectID,tax)%>%
  dplyr::summarize(Abundance=sum(Abundance), .groups = "drop")%>%
  group_by(Treatment,tax)%>%
  dplyr::summarize(Mean = mean(Abundance), .groups = "drop")%>%
  arrange(-Mean)

#Show top
top

top10 <- top$tax[1:100]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))

df0 <- df0[order(df0$Sample, decreasing = TRUE), ]

#Set order for samples
df0$Sample <- factor(df0$Sample, levels = c("FVT_ChP", "FVT_SDT", "FVT_PyT", "FVT_UnT", "HFD_control", "LFD_control"))

df0$Abundance <- df0$Abundance*100


barplot.genus.treatment <- ggplot(df0, aes(Sample, Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(width = 0.8) +
  scale_fill_jco(name = "Taxonomy") +
  scale_fill_manual(values=rep(col_fil,5),name="Phylum")+
  scale_x_discrete(breaks = c("FVT_ChP", "FVT_SDT", "FVT_PyT", "FVT_UnT", "HFD_control", "LFD_control"),
                   labels = c("FVT-ChP", "FVT-SDT", "FVT-PyT", "FVT-UnT", "HFD-control", "LFD-control"))+
  theme_classic() +
  theme(text = element_text(size = 10, colour = "Black"),
        axis.line=element_line(size=0.5),
        #panel.border = element_blank(),
        axis.text=element_text(size = 10, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(angle = 0, size=12, face = "bold"),
        legend.text = element_text(face = 
                                     "italic"),
        legend.key.size = unit(10, "pt")
  ) +
  # scale_fill_discrete(name = NULL)+
  ylab("Mean Relative abundance (%)") +
  xlab("Treatment")

barplot.genus.treatment

```








```{r}

PSBamp <- phyloseq_to_ampvis2(ps0)  

#Bacteriome

amp_heatmap(PSBamp,
            group_by = "Treatment",
            facet_by = "Sample_time_point",
            plot_values = FALSE,
            tax_show = 30,
            tax_aggregate = "Genus" ,
            #tax_add = "Family",
            tax_empty = "best",
            normalise = TRUE,
            color_vector = c("white", "red"),
            plot_colorscale = "sqrt",
            plot_legendbreaks = c(1, 10, 25, 50)
) +
  theme_classic() +
  # facet_wrap(~Sample_time_point)+
  theme(strip.background = element_rect(colour = "black", fill = "white"),
        strip.text.x = element_text(face = "bold"),
        axis.text.x=element_text(angle = -45, hjust = 0,vjust = 0.2))
```



##Barplots based on treatments. 
```{r, eval = T, include = T, echo = F, message = F}

  vir.phyl <- tax_glom(PSB, "Genus", NArm = FALSE)

ps0 <- transform_sample_counts(vir.phyl, function(x) x / sum(x))

ps1 <- merge_samples(ps0, "Treatment")

ps1 <- transform_sample_counts(ps1, function(x) x / sum(x))


#Create melted dataframe
df <- psmelt(ps1)

#Select last non-empty taxonomic rank
df[df==""] <- NA

df$tax <- apply(df, 1, function(x) tail(na.omit(x), 1))

#Arrange samples by mean abundance
top <- df %>%
  group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  arrange(-Mean)
#Show top
top


top<-df %>%
  # filter(level=="Phylum")%>%
  group_by(Treatment,Sample,tax)%>%
  dplyr::summarize(Abundance=sum(Abundance), .groups = "drop")%>%
  group_by(Treatment,tax)%>%
  dplyr::summarize(Mean = mean(Abundance), .groups = "drop")%>%
  arrange(-Mean)



top10 <- top$tax[1:50]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))

df0 <- df0[order(df0$Sample, decreasing = TRUE), ]

#Set order for samples
#df0$Treatment <- factor(df0$Treatment, levels = c("Control", "Amp", "FOP", "LM", "LM+Amp", "LM+FOP"))

df$Sample <- factor(df0$Sample, levels = c("3", "4", "5"))

barplot.genus.treatment <- ggplot(df0, aes(Sample, Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(width = 0.8) +
  scale_fill_jco(name = "Taxonomy") +
  scale_fill_manual(values=rep(col_fil,5),name="Genus")+
  theme_classic() +
  theme(text = element_text(size = 10, colour = "Black"),
        axis.line=element_line(size=0.5),
        #panel.border = element_blank(),
        axis.text=element_text(size = 10, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(angle = 0, size=12, face = "bold"),
        #legend.position = "none"
  ) +
  ylab("Relative abundance") +
  xlab("Treatment")

barplot.genus.treatment

```





## Termination: Lactococcus- barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}
PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Lactococcus"), "Genus", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
# df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df_Lactococcus <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_Lactococcus_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Lactococcus") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_Lactococcus_species
```

## abundance significant calculation
```{r}
PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))


ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Clostridium"), "Genus", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
# df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df_Clostridium <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


write.csv(df_Lactobacillus,"Mao_output_file/Abundance/Lactobacillus.csv")
write.csv(df_Bacteroides,"Mao_output_file/Abundance/Bacteroides.csv")
write.csv(df_Staphylococcus,"Mao_output_file/Abundance/Staphylococcus.csv")
write.csv(df_Bifidobacterium,"Mao_output_file/Abundance/Bifidobacterium.csv")
write.csv(df_Oscillospira,"Mao_output_file/Abundance/Oscillospira.csv")
write.csv(df_Ruminococcus,"Mao_output_file/Abundance/Ruminococcus.csv")
write.csv(df_Lactococcus,"Mao_output_file/Abundance/Lactococcus.csv")
write.csv(df_Dehalobacterium,"Mao_output_file/Abundance/Dehalobacterium.csv")
write.csv(df_Allobaculum,"Mao_output_file/Abundance/Allobaculum.csv")
```





## Termination: Lactococcus - Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Lactococcus abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Lactococcus abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Lactococcus abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```


## Termination: Lactobacillus - barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Lactobacillus"), "Genus", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_Lactobacillus_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Lactobacillus") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_Lactobacillus_species
```
## Termination: Lactobacillus -Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Lactobacillus abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Lactobacillus abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Lactobacillus abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```

## Termination: Bacteroides - barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Bacteroides"), "Genus", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_Bacteroides_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Bacteroides") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_Bacteroides_species
```
## Termination: Bacteroides -Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Bacteroides abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Bacteroides abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Bacteroides abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```

## Termination: Allobaculum - barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Allobaculum"), "Genus", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_Allobaculum_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Allobaculum") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_Allobaculum_species
```


## Termination: Allobaculum -Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Allobaculum abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Allobaculum abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Allobaculum abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```

## Termination: Bifidobacterium - barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Bifidobacterium"), "Genus", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_Bifidobacterium_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Bifidobacterium") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_Bifidobacterium_species
```


## Termination: Bifidobacterium -Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Bifidobacterium abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Bifidobacterium abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Bifidobacterium abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```
## Termination: f__Clostridiaceae_zOTU_52 - barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Genus=="Clostridiaceae_zOTU_52"), "Genus", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Genus)] <- paste0(df$tax[is.na(df$Genus)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_f__Clostridiaceae_zOTU_52_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "f__Clostridiaceae_zOTU_52") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_f__Clostridiaceae_zOTU_52_species
```


## Termination: f__Clostridiaceae_zOTU_52 -Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on f__Clostridiaceae_zOTU_52 abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on f__Clostridiaceae_zOTU_52 abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on f__Clostridiaceae_zOTU_52 abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```

```{r, eval = T, include = T, echo = F, message = F}
# Boxplots for Termination

#bac.phyl <- tax_glom(subset_samples(PSB, Sampling.Nation=="Malta"), "Genus", NArm = FALSE)

bac.phyl <- tax_glom(PSB, "Genus", NArm = FALSE)


ps0 <- transform_sample_counts(bac.phyl, function(x) x / sum(x))

#Select desired rank
df <- psmelt(ps0)

#Select last non-empty taxonomic rank
df[df==""] <- NA

df$tax <- apply(df, 1, function(x) tail(na.omit(x), 1))

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  arrange(-Mean)
#Show top
top

top10 <- top$tax[1:110]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))

bar_bac_genus_boxplot <- ggplot(df0, aes(Treatment, Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_boxplot(show.legend = FALSE) +
  scale_fill_jco() +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=90, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  scale_y_continuous(breaks=c(0.00,0.25,0.5)) +
  xlab("") +
  ylab("Relative abundance") +
  facet_grid(tax~.,scales="free") 

bar_bac_genus_boxplot
```

# Abundance heatmaps

## Bacteria  - Treatment -4 timepoint Phylum
If want to change it to 3 time point you can manually delete one time

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE}
vir.phyl <- tax_glom(PSB, "Phylum", NArm = FALSE)

ps0 <- transform_sample_counts(vir.phyl, function(x) x / sum(x))

#Load phyloseq files to ampvis2 format

PSBamp <- phyloseq_to_ampvis2(ps0)




#Bacteriome


amp_heatmap(PSBamp,
            group_by = "Treatment",
            facet_by = "Sample_time_point",
            plot_values = FALSE,
            tax_show = 20,
            tax_aggregate = "Phylum" ,
            # tax_add = "Family",
            tax_empty = "best",
            normalise = TRUE,
            color_vector = c("white", "red"),
            plot_colorscale = "sqrt",
            plot_legendbreaks = c(1, 10, 25, 50)
) +
  theme_classic() +
  # facet_wrap(~Sample_time_point)+
  theme(strip.background = element_rect(colour = "black", fill = "white"),
        strip.text.x = element_text(face = "bold", size = 10),
        axis.text.x=element_text(angle = -45, hjust = 0,vjust = 0.2,size = 10),
        axis.text.y=element_text(size = 10),
        axis.title.y=element_text(size = 10))
```

### Termination: Firmicutes- barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}
PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Phylum=="Firmicutes"), "Phylum", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Phylum)] <- paste0(df$tax[is.na(df$Phylum)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top
top_Firmicutes <- top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_Firmicutes_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Firmicutes") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_Firmicutes_species
```

### Termination: Firmicutes - Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Firmicutes abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Firmicutes abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Firmicutes abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```

### Termination: Bacteroidetes- barplot on specific species/genus

```{r, eval = T, include = T, echo = F, message = F}
PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))
################################## barplots
################By species

PSB.rel <- transform_sample_counts(PSB_Termination, function(x) x / sum(x))

ps0 <- tax_glom(subset_taxa(PSB.rel, Phylum=="Bacteroidetes"), "Phylum", NArm = FALSE)


#Select desired rank
df <- psmelt(ps0)

#Replace missing values with NA
df[df==""] <- NA

#Select last non-empty taxonomic rank
df$tax <- apply(df, 1, function(x) tail(stats::na.omit(x), 1))

#Add "unknown" to taxonomy columns where genus is not classified 
df$tax[is.na(df$Phylum)] <- paste0(df$tax[is.na(df$Phylum)],"unknown genus")
#Add "unknown" to taxonomy columns where Species is not classified 
df$tax[is.na(df$Species)] <- paste0(df$tax[is.na(df$Species)]," unknown species")

#Arrange smaples by mean abundance
top <- df %>%
  dplyr::group_by(Sample, tax) %>%
  dplyr::summarize(Mean = mean(Abundance)) %>%
  dplyr::arrange(-Mean)
#Show top
top
top_Bacteroidetes <- top

top10 <- top$tax[1:120]
df0 <- df %>%
  mutate(tax = fct_other(tax, c(as.matrix(top10))))


bar_Bacteroidetes_species <- ggplot(df0, aes(x=Treatment, y=Abundance, fill = fct_reorder(tax, Abundance, .fun = mean))) + 
  geom_col(show.legend = TRUE) +
  scale_fill_jco(name = "Bacteroidetes") +
  theme_classic() +
  theme(
    #axis.line=element_line(size=1), 
    axis.text=element_text(size=10, colour = "Black"),
    axis.text.x=element_text(angle=45, hjust = 1, size=10),
    axis.ticks=element_line(colour = "Black"),
    strip.text.y = element_text(angle = 0,face = "italic"),
    #strip.text.x = element_text(face = "italic",size=30, family = "sans"),
    strip.background = element_blank(),
    text = element_text(size = 10)
  ) +
  #scale_y_continuous(breaks=c(0.00,0.05,0.1,0.15)) +
  xlab("") +
  ylab("Relative abundance") 
bar_Bacteroidetes_species
```


### Termination: Bacteroidetes - Compared specific species/genus with control group - calculation

```{r, eval = T, include = T, echo = F, message = F}

#Effect of FVT on Bacteroidetes abundance
df$Treatment <- fct_relevel(df$Treatment, "LFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Bacteroidetes abundance
df$Treatment <- fct_relevel(df$Treatment, "HFD_control") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

#Effect of FVT on Bacteroidetes abundance
df$Treatment <- fct_relevel(df$Treatment, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Abundance ~Treatment, data = df)
summary(fit_norm)

```
### Bacteroidetes/Firmicutes
```{r}
# write.csv(top_Bacteroidetes,"Bacteroidetes_phylum.csv")
# write.csv(top_Firmicutes,"Firmicutes_phylum.csv")
top_ratio <- read.csv("Mao_output_file/Bacteroidetes divided by Firmicutes.csv")
top_ratio

#Effect of FVT on Bacteroidetes abundance
top_ratio$Column2 <- fct_relevel(top_ratio$Column2, "LFD_control") #Releveling to control 
fit_norm <- lm(Ratio ~Column2, data = top_ratio)
summary(fit_norm)

#Effect of FVT on Bacteroidetes abundance
top_ratio$Column2 <- fct_relevel(top_ratio$Column2, "HFD_control") #Releveling to control 
fit_norm <- lm(Ratio ~Column2, data = top_ratio)
summary(fit_norm)

#Effect of FVT on Bacteroidetes abundance
top_ratio$Column2 <- fct_relevel(top_ratio$Column2, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Ratio ~Column2, data = top_ratio)
summary(fit_norm)

top_ratio$Column2 <- factor(top_ratio$Column2, levels = c("FVT_ChP", "FVT_SDT", "FVT_PyT", "FVT_UnT", "HFD_control", "LFD_control"))

means <- aggregate(Ratio ~  Column2, top_ratio, mean)

# ggplot(top_ratio, aes(x=Column2, y=Ratio, color=Column2)) +
#     geom_boxplot(alpha=0.1, lwd = 0.5)+
#     # scale_color_jco() +
#     stat_boxplot(geom = "errorbar", width = 0.15) +
#     # facet_wrap(~Column2, nrow = 1, ncol = 4)+             
#     theme_classic() +
#     stat_summary(fun=means, colour="darkred", geom="point", 
#                shape=18, size=3, show.legend=FALSE) + 
#    geom_text(data = means, aes(label = Column2, y = Ratio + 0.08))+
#     theme(text = element_text(size = 8),
#         axis.line=element_line(size=1),
#         axis.text=element_text(size = 8, colour = "Black"),
#         axis.ticks=element_line(size=1, colour = "Black"),
#         strip.background = element_rect(colour = "white", fill = "white"),
#         axis.text.x=element_text(angle = 45, hjust = 1),
#         strip.text.x = element_text(size = 8, angle = 0),
#         legend.text = element_text(size = 8)
#   ) +
#       scale_fill_manual(values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
#                     breaks = c( "FVT_ChP", "FVT_SDT","FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
#                     labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
#                   )+
#     ggtitle("Ratio of Bacteroidetes/Firmicutes
# ") +
#     xlab("")+
#     ylab("Ratio")

ggplot(top_ratio,aes(x=Column2,y=Ratio)) +
geom_boxplot(aes(fill=Column2)) +
stat_boxplot(geom = "errorbar", width = 0.15) +
stat_summary(fun.y = mean, geom="point",colour="Black", size=3) +
  geom_text(data = means, aes(label = Ratio, y = Ratio + 0.08))+
stat_summary(fun.data = means, geom="text", vjust=-0.7)+
  theme(text = element_text(size = 8),
        axis.line=element_line(size=1),
        axis.text=element_text(size = 8, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        panel.background = element_rect(fill='white'),
        plot.background = element_rect(fill='white', color=NA),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(size = 8, angle = 0),
        legend.text = element_text(size = 8)
  ) +
  scale_fill_manual(values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT","FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )+
  ggtitle("Ratio of Bacteroidetes/Firmicutes
") +
    xlab("")+
    ylab("Ratio")




```

```{r}

```





## Bacteria  - Treatment -4 timepoint _Genus
If want to change it to 3 time point you can manually delete one time

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE}
vir.phyl <- tax_glom(PSB, "Genus", NArm = FALSE)

ps0 <- transform_sample_counts(vir.phyl, function(x) x / sum(x))

#Load phyloseq files to ampvis2 format

PSBamp <- amp_load(ps0)




#Bacteriome


amp_heatmap(PSBamp,
            group_by = "Treatment",
            facet_by = "Sample_time_point",
            plot_values = FALSE,
            tax_show = 20,
            tax_aggregate = "Genus" ,
            tax_add = "Family",
            tax_empty = "best",
            normalise = TRUE,
            color_vector = c("white", "red"),
            plot_colorscale = "sqrt",
            plot_legendbreaks = c(1, 10, 25, 50)
) +
  theme_classic() +
  mytheme_abundance
  # facet_wrap(~Sample_time_point)+
  theme(strip.background = element_rect(colour = "black", fill = "white"),
        strip.text.x = element_text(face = "bold", size = 10),
        axis.text.x=element_text(angle = -45, hjust = 0,vjust = 0.2,size = 10),
        axis.text.y=element_text(size = 10),
        axis.title.y=element_text(size = 10))
```

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE}
vir.phyl <- tax_glom(PSB, "Species", NArm = FALSE)

ps0 <- transform_sample_counts(vir.phyl, function(x) x / sum(x))

#Load phyloseq files to ampvis2 format


PSBamp <- phyloseq_to_ampvis2(ps0)




#Bacteriome


amp_heatmap(PSBamp,
            group_by = "Treatment",
            facet_by = "Sample_time_point",
            plot_values = FALSE,
            tax_show = 20,
            tax_aggregate = "Species" ,
            # tax_add = "Family",
            tax_empty = "best",
            normalise = TRUE,
            color_vector = c("white", "red"),
            plot_colorscale = "sqrt",
            plot_legendbreaks = c(1, 10, 25, 50)
) +
  theme_classic() +
  # facet_wrap(~Sample_time_point)+
  theme(strip.background = element_rect(colour = "black", fill = "white"),
        strip.text.x = element_text(face = "bold", size = 10),
        axis.text.x=element_text(angle = -45, hjust = 0,vjust = 0.2,size = 10),
        axis.text.y=element_text(size = 10),
        axis.title.y=element_text(size = 10))
```



PSB_Arrival <- subset_samples(PSB, Sample_time_point %in% c("Arrival"))

## Bacteria  - Treatment -4 timepoint _Species
If want to change it to 3 time point you can manually delete one time
##Lda score
```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE}
library(MASS)
PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

vir.phyl <- tax_glom(PSB_Termination, "Species", NArm = FALSE)

ps0_termination<- transform_sample_counts(vir.phyl, function(x) x / sum(x))

data(GlobalPatterns)

# 
# otu_table <- as.data.frame(otu_table(ps0_termination))
# tax_table <- as.data.frame(tax_table(ps0_termination))

# # Merge the two tables by the OTU ID
# otu_tax_table <- merge(otu_table, tax_table, by = "row.names")
# 
# # Rename the first column to "OTU ID"
# colnames(otu_tax_table)[1] <- "OTU ID"
# 
# # Write the merged table to a file
# write.table(otu_tax_table, file = "otu_tax_table_abundance_termination.txt", sep = "\t", quote = FALSE, row.names = FALSE)

# Extract the OTU table and sample data
otu_table <- otu_table(ps0_termination)
sample_data <- sample_data(ps0_termination)

# Identify constant variables within each group
variances <- apply(otu_table, 2, function(x) tapply(x, sample_data$Treatment, var))
constant_vars <- which(colSums(variances == 0) > 0)

# Convert the OTU table to a matrix
otu_table_filtered <- otu_table[,-constant_vars]
otu_matrix <- as.matrix(otu_table)

# Perform the LDA analysis
lda_model <- lda(t(otu_matrix), sample_data$Treatment)

# Compute the LDA scores
lda.scores <- predict(lda(t(otu_table)), sample_data)$x

```



# Heatmap base on deseq - host - Termination

```{r}
# load data to calculate the matrix
ps.rel <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

metadata <- data.frame(sample_data(ps.rel),check.names = FALSE)

abun_matri <- data.frame(otu_table(ps.rel),check.names = FALSE)
abun_matri_t <- t(abun_matri)
library(reshape2)
rel_tab_l <- melt(abun_matri_t,variable.name = "mags")
colnames(rel_tab_l) <- c("sample","mags","abundance")

# match the Treatment information
index <- match(rel_tab_l$sample, rownames(metadata))
table(is.na(index))
rel_tab_l$Treatment <- metadata$Treatment[index]




library(rstatix)
#1
tab_ChP_LFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_ChP", "Lean_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) %>%
  adjust_pvalue(method = "fdr")%>%
  add_significance("p.adj")
#2
tab_SDT_LFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_SDT", "Lean_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#3
tab_PyT_LFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_PyT", "Lean_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#4
tab_UnT_LFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_UnT", "Lean_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#5
tab_HFD_LFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("HFD_control", "Lean_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#6
tab_ChP_HFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_ChP", "HFD_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 

#7
tab_SDT_HFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_SDT", "HFD_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#8
tab_PyT_HFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_PyT", "HFD_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#9
tab_UnT_HFD_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_UnT", "HFD_control") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#10
tab_ChP_UnT_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_ChP", "FVT_UnT") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) #%>%
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 

#11
tab_SDT_UnT_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_SDT", "FVT_UnT") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment) 
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 
#12
tab_PyT_UnT_all <- rel_tab_l %>%
  filter(Treatment %in% c("FVT_PyT", "FVT_UnT") ) %>%
  group_by(mags) %>%
  kruskal_test(abundance~Treatment)
  # adjust_pvalue(method = "BH")%>%
  # add_significance("p.adj") 


#Include wilcoxon p.adj FMT2 vs CON, FMT2 vs FMT1, taxonomy for rows
tab_ChP_LFD <- subset(tab_ChP_LFD_all, p < 0.05)
tab_SDT_LFD <-subset(tab_SDT_LFD_all, p < 0.05)
tab_PyT_LFD <- subset(tab_PyT_LFD_all, p < 0.05)
tab_UnT_LFD <- subset(tab_UnT_LFD_all, p < 0.05)
tab_HFD_LFD <-subset(tab_HFD_LFD_all, p < 0.05)
tab_ChP_HFD <- subset(tab_ChP_HFD_all, p < 0.05)
tab_SDT_HFD <- subset(tab_SDT_HFD_all, p < 0.05) #no difference
tab_PyT_HFD <-subset(tab_PyT_HFD_all, p < 0.05)  #no difference
tab_UnT_HFD <- subset(tab_UnT_HFD_all, p < 0.05) #no difference
tab_ChP_UnT <- subset(tab_ChP_UnT_all, p < 0.05) #no difference
tab_SDT_UnT <-subset(tab_SDT_UnT_all, p < 0.05)  #no difference
tab_PyT_UnT <- subset(tab_PyT_UnT_all, p < 0.05) #no difference
ASV <- unique(c(as.character(tab_ChP_LFD$mags), #1
                as.character(tab_SDT_LFD$mags), #2
                as.character(tab_PyT_LFD$mags), #3
                as.character(tab_UnT_LFD$mags), #4
                as.character(tab_HFD_LFD$mags), #5
                as.character(tab_ChP_HFD$mags), #6
                as.character(tab_SDT_HFD$mags), #7
                as.character(tab_PyT_HFD$mags),#8
                as.character(tab_UnT_HFD$mags), #9
                as.character(tab_ChP_UnT$mags),#10
                as.character(tab_SDT_UnT$mags), #11
                as.character(tab_PyT_UnT$mags)))#12

ps.sample.rel <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

# 
# #only take the donors belong mags
# tax <- data.table::as.data.table(tax_table(ps.sample.rel),keep.rownames = TRUE)
# index <- as.character(tax$Genus) 
# tax_lac <- tax[index,]$rn 
# ps.sample.rel.donor <- prune_taxa(tax_lac,ps.sample.rel)
# ps.sample.rel.donor
# ps.sample.rel.test <- readRDS("ps_denovo_mags_donorSource.rds")
# tax.ttest<- data.table::as.data.table(tax_table(ps.sample.rel.test),keep.rownames = TRUE)
# A<-ps.sample.rel@tax_table
#different by p different colonized in FMT1 and FMT2

Vector1 <- as.character(tab_ChP_LFD$mags)
# Vector2 <- as.character(tab_SDT_LFD$mags)
# Vector3 <- as.character(tab_PyT_LFD$mags)
Vector4 <- as.character(tab_UnT_LFD$mags)
Vector5 <- as.character(tab_HFD_LFD$mags)
Vector6 <- as.character(tab_ChP_HFD$mags)
# Vector7 <- as.character(tab_SDT_HFD$mags)
# Vector8 <- as.character(tab_PyT_HFD$mags)
Vector9 <- as.character(tab_UnT_HFD$mags)
Vector10 <- as.character(tab_ChP_UnT$mags)
# Vector11 <- as.character(tab_SDT_UnT$mags)
# Vector12 <- as.character(tab_PyT_UnT$mags)
# merge_vector <- unique(c(Vector1,Vector2,Vector3,Vector4,Vector5,Vector6,Vector7,Vector8,Vector9,Vector10,Vector11,Vector12))
merge_vector <- unique(c(Vector1,Vector4,Vector5,Vector6,Vector9,Vector10))

ps.sample.rel.sig <- prune_taxa(merge_vector,ps.sample.rel)
ps.sample.rel.sig

otu_abun_select <- data.frame(otu_table(ps.sample.rel.sig), check.names = F)
#import relavant metadata
metadata <- sample_data(ps.sample.rel.sig)
tax.clean <- data.frame(tax_table(ps.sample.rel.sig))
# use complex heatmap for visualization, multiple row annotation
# extract the relative abundance matrix for thoes responding signals

library(ComplexHeatmap)
library(dplyr)
library(circlize)
# create a variable to define the subgroup, "FVT_ChP","FVT_SDT","FVT_PyT","FVT_UnT","HFD_control","LFD_control" ...
# order the matrix by the subgroup
metadata$Treatment <- factor(metadata$Treatment, levels = c("FVT_ChP","FVT_SDT","FVT_PyT","FVT_UnT","HFD_control","LFD_control"))
meta_order <- metadata[order(metadata$Treatment),]
# re_order the col
mat <- otu_abun_select
mat <- as.matrix(mat[,rownames(meta_order)])



base_mean = rowMeans(mat)
mat_scaled = t(scale(t(mat)))
# calculate heatmap annotation
tax_heatmap <- tax.clean[rownames(mat_scaled),]
tax_heatmap$tab_ChP_LFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_ChP_LFD$mags,"*","ns"))
tax_heatmap$tab_SDT_LFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_SDT_LFD$mags,"*","ns"))
tax_heatmap$tab_PyT_LFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_PyT_LFD$mags,"*","ns"))
tax_heatmap$tab_UnT_LFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_UnT_LFD$mags,"*","ns"))
tax_heatmap$tab_HFD_LFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_HFD_LFD$mags,"*","ns"))
tax_heatmap$tab_ChP_HFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_ChP_HFD$mags,"*","ns"))
tax_heatmap$tab_SDT_HFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_SDT_HFD$mags,"*","ns"))
tax_heatmap$tab_PyT_HFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_PyT_HFD$mags,"*","ns"))
tax_heatmap$tab_UnT_HFD <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_UnT_HFD$mags,"*","ns"))
tax_heatmap$tab_ChP_UnT <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_ChP_UnT$mags,"*","ns"))
tax_heatmap$tab_SDT_UnT <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_SDT_UnT$mags,"*","ns"))
tax_heatmap$tab_PyT_UnT <- sapply(rownames(tax_heatmap), function(x) ifelse(x %in% tab_PyT_UnT$mags,"*","ns"))


#tab_ChP_LFD
index <- match(rownames(tax_heatmap), tab_ChP_LFD_all$mags)
index
tax_heatmap$ChP_LFD_p <- tab_ChP_LFD_all$p[index]
tax_heatmap$ChP_LFD_p[tax_heatmap$ChP_LFD_p=="NaN"] <- 1

#tab_SDT_LFD
index <- match(rownames(tax_heatmap), tab_SDT_LFD_all$mags)
index
tax_heatmap$SDT_LFD_p <- tab_SDT_LFD_all$p[index]
tax_heatmap$SDT_LFD_p[tax_heatmap$SDT_LFD_p=="NaN"] <- 1



tax_heatmap <- tax_heatmap[order(tax_heatmap$ChP_LFD_p,
                                 tax_heatmap$SDT_LFD_p,
                                 tax_heatmap$PyT_LFD_p,
                                 tax_heatmap$UnT_LFD_p,
                                 tax_heatmap$HFD_LFD_p,
                                 tax_heatmap$ChP_HFD_p,
                                 tax_heatmap$SDT_HFD_p,
                                 tax_heatmap$PyT_HFD_p,
                                 tax_heatmap$UnT_HFD_p,
                                 tax_heatmap$ChP_UnT_p,
                                 tax_heatmap$SDT_UnT_p,
                                 tax_heatmap$PyT_UnT_p),]


#note: PyT_UnT_p replaced FMT2vsCON_p

# sorted_indice <- order(tax_heatmap$Species)
# sorted_indice
# tax_heatmap <-tax_heatmap[sorted_indice,]

# my_palette <- c("#a6cee3","#1f78b4","#b2df8a","#33a02c","#fb9a99","#e31a1c","#fdbf6f","#ff7f00","#cab2d6","#6a3d9a","#ffff99","darkblue")
# 
# my_palette <- c("darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen", "deeppink", "khaki2", "firebrick", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey", "darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen", "deeppink", "khaki2", "firebrick", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey")
# 
my_palette <- c("orangered4","red","purple","maroon1","lawngreen","blue","orange1","wheat2","violetred3","turquoise3","tomato2","thistle2","tan2","steelblue4","springgreen4","snow4","yellow2","slategray2","slateblue4","sienna4","orchid4","royalblue3","cyan1","darkmagenta","burlywood4","black")

library(RColorBrewer)
# colors <- brewer.pal(12, "Paired")
# more_colors <- brewer.pal(10, "Set3")
# all_colors <- c(colors, more_colors)
# my_palette <- rep(all_colors, length.out = 30)

tax_heatmap$genus_p <- stringr::str_replace(as.character(tax_heatmap$Genus), " ","_")
tax_heatmap$genus_p <- stringr::str_replace(as.character(tax_heatmap$Genus), "_","")

genus <- unique(as.character(tax_heatmap$genus_p))
genus_col <- colorRampPalette(my_palette[1:27])(length(genus))
names(genus_col) <- genus

# tax_heatmap$family_p <- stringr::str_replace(as.character(tax_heatmap$Family), " ","_")
# tax_heatmap$family_p <- stringr::str_replace(as.character(tax_heatmap$Family), "_","")
# 
# family <- unique(as.character(tax_heatmap$family_p))
# family_col <- colorRampPalette(my_palette[1:27])(length(family))
# names(family_col) <- family





# magSource <- unique(as.character(tax_heatmap$source))
# magSource_col <- colorRampPalette(my_palette[1:12])(length(magSource))
# names(magSource_col) <- magSource
pvalue_col_fun = colorRamp2(c(0, 1, 2), c("lightseagreen", "white", "red"))
# ha_row <- HeatmapAnnotation(CHPvsLFD=anno_simple(-log10(tax_heatmap$ChP_LFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_ChP_LFD,"ns")),
#                             SDTvsLFD=anno_simple(-log10(tax_heatmap$SDT_LFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_SDT_LFD,"ns")),
#                             PYTvsLFD=anno_simple(-log10(tax_heatmap$PyT_LFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_PyT_LFD,"ns")),
#                             UNTvsLFD=anno_simple(-log10(tax_heatmap$UnT_LFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_UnT_LFD,"ns")),
#                             HFDvsLFD=anno_simple(-log10(tax_heatmap$HFD_LFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_HFD_LFD,"ns")),
#                             CHPvsHFD=anno_simple(-log10(tax_heatmap$ChP_HFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_ChP_HFD,"ns")),
#                             SDTvsHFD=anno_simple(-log10(tax_heatmap$SDT_HFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_SDT_HFD,"ns")),
#                             PYTvsHFD=anno_simple(-log10(tax_heatmap$PyT_HFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_PyT_HFD,"ns")),
#                             UNTvsHFD=anno_simple(-log10(tax_heatmap$UnT_HFD_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_UnT_HFD,"ns")),
#                             CHPvsUNT=anno_simple(-log10(tax_heatmap$ChP_UnT_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_ChP_UnT,"ns")),
#                             SDTvsUNT=anno_simple(-log10(tax_heatmap$SDT_UnT_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_SDT_UnT,"ns")),
#                             PYTvsUNT=anno_simple(-log10(tax_heatmap$PyT_UnT_p),col = pvalue_col_fun,pch = na_if(tax_heatmap$tab_PyT_UnT,"ns")),
#                             Class=anno_simple(tax_heatmap$class_p, col = class_col),
#                             which = "row")

ha_row <- HeatmapAnnotation(CHPvsLFD=anno_simple(-log10(tax_heatmap$ChP_LFD_p),col = pvalue_col_fun),
SDTvsLFD=anno_simple(-log10(tax_heatmap$SDT_LFD_p),col = pvalue_col_fun),
PYTvsLFD=anno_simple(-log10(tax_heatmap$PyT_LFD_p),col = pvalue_col_fun),
UNTvsLFD=anno_simple(-log10(tax_heatmap$UnT_LFD_p),col = pvalue_col_fun),
HFDvsLFD=anno_simple(-log10(tax_heatmap$HFD_LFD_p),col = pvalue_col_fun),
CHPvsHFD=anno_simple(-log10(tax_heatmap$ChP_HFD_p),col = pvalue_col_fun),
SDTvsHFD=anno_simple(-log10(tax_heatmap$SDT_HFD_p),col = pvalue_col_fun),
PYTvsHFD=anno_simple(-log10(tax_heatmap$PyT_HFD_p),col = pvalue_col_fun),
UNTvsHFD=anno_simple(-log10(tax_heatmap$UnT_HFD_p),col = pvalue_col_fun),
CHPvsUNT=anno_simple(-log10(tax_heatmap$ChP_UnT_p),col = pvalue_col_fun),
SDTvsUNT=anno_simple(-log10(tax_heatmap$SDT_UnT_p),col = pvalue_col_fun),
PYTvsUNT=anno_simple(-log10(tax_heatmap$PyT_UnT_p),col = pvalue_col_fun),
Species=anno_simple(tax_heatmap$genus_p, col = genus_col),
which = "row")
meta_order$Treatment <- factor(meta_order$Treatment, levels = c("FVT_ChP","FVT_SDT","FVT_PyT","FVT_UnT","HFD_control","LFD_control"), labels = c("FVT-ChP","FVT-SDT","FVT-PyT","FVT-UnT","HFD-control","LFD-control"))


ha_col = HeatmapAnnotation(Treatment=meta_order$Treatment,col = list(Treatment=c("FVT-ChP"="#F27970", "FVT-SDT"="#BB9727", "FVT-PyT"="#54B345", "FVT-UnT"="#32B897", "HFD-control"="#05B9E2","LFD-control"="#8983BF")))

# right_row <- rowAnnotation(foo = anno_text(tax_heatmap$Species))


# mat_scaled<-mat_scaled[rownames(tax_heatmap), ]


# sorted_indice <- order(tax_heatmap$Species)
# sorted_indice
# tax_heatmap <-tax_heatmap[sorted_indice,]

Hist <- Heatmap(mat_scaled[rownames(tax_heatmap),] , cluster_columns = FALSE, cluster_rows = TRUE,
                name="Z-score", col=colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white","deeppink3")),
                top_annotation = ha_col, 
                left_annotation = ha_row,
                show_row_names = FALSE, show_column_names = FALSE,
                heatmap_legend_param = list(legend_direction = "horizontal"))
# +
                # rowAnnotation(species = anno_text(tax_heatmap$Species, just = "left", location = unit(0.5, "npc"), show_name = TRUE), annotation_name_rot = 0)

Hist

# define the two legend
lgd_group = Legend(title = "Treatment", legend_gp = gpar(fill = c("#F27970", "#BB9727", "#54B345", "#32B897", "#05B9E2","#8983BF")),
                   labels = c("FVT-ChP","FVT-SDT","FVT-PyT","FVT-UnT","HFD-control","LFD-control"))

lgd_genus = Legend(title = "Genus", legend_gp = gpar(fill = genus_col),labels = genus, ncol = 2)

lgd_sig = Legend(title= " ", pch = "*", type = "points", labels = "p < 0.05")
lgd_pvalue = Legend(title = "p value",
                    col_fun  = pvalue_col_fun,
                    at = c(0, 1, 2, 3, 4),
                    labels = c("1", "0.1", "0.05", "0.01", "0.001"),
                    direction = "horizontal")

draw(Hist, heatmap_legend_list=list(lgd_group, lgd_genus,
                                    lgd_pvalue,lgd_sig),
     heatmap_legend_side = "bottom", annotation_legend_side = "bottom")

# write.csv(matscaled,"virome deseq heatmaptesting/star annotation.csv")
# tax_heatmap<-tax_heatmap[rownames(mat_scaled_res), ]


```



#Deseq2 - Termination- Genus_FVT_ChP vs LFD 
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_ChP","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] 
#Select values from results file with adj p values>alpha AND log2fold chnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB.pseu)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()





```


#Deseq2 - Termination- Genus_FVT_ChP vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_ChP","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

#Deseq2 - Termination- Genus_FVT_ChP vs FVT_UnT
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_ChP","FVT_UnT"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - Termination- Genus_FVT_SDT vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_SDT","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - Termination- Genus_FVT_SDT vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_SDT","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

#Deseq2 - Termination- Genus_FVT_SDT vs FVT_UnT
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_SDT","FVT_UnT"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - Termination- Genus_FVT_PyT vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_PyT","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - Termination- Genus_FVT_PyT vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_PyT","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

#Deseq2 - Termination- Genus_FVT_PyT vs FVT_UnT
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_PyT","FVT_UnT"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB_Termination)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - Termination- Genus_FVT_UnT vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_UnT","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - Termination- Genus_FVT_UnT vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_UnT","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

#Deseq2 - Termination- Genus_HFD vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("HFD_control","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT - Genus_FVT_ChP vs LFD 
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_ChP","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_ChP vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_ChP","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

#Deseq2 - 1w_after_FVT- Genus_FVT_ChP vs FVT_UnT
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_ChP","FVT_UnT"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_SDT vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_SDT","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_SDT vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_SDT","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_SDT vs FVT_UnT
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_SDT","FVT_UnT"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_PyT vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_PyT","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_PyT vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_PyT","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

#Deseq2 - 1w_after_FVT- Genus_FVT_PyT vs FVT_UnT
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_PyT","FVT_UnT"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_UnT vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_UnT","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```


#Deseq2 - 1w_after_FVT- Genus_FVT_UnT vs HFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("FVT_UnT","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

#Deseq2 - 1w_after_FVT- Genus_HFD vs LFD
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))

PSB.pseu <- subset_samples(PSB_Termination, Treatment %in% c("HFD_control","LFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
# Species order based on abundance 
# x = tapply(sigtab$log2FoldChange, sigtab$Species, function(x) max(x))
# x = sort(x, TRUE)
# sigtab$Species = factor(as.character(sigtab$Species), levels=names(x))


diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            # tax_add = "Family"
            ) +
  theme_classic()

```

PSB_Arrival <- subset_samples(PSB, Sample_time_point %in% c("Arrival"))

## Bacteria  - Treatment -Termination
```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE}
#Load phyloseq files to ampvis2 format
ps0_termination <- subset_samples(ps0, Sample_time_point %in% c("Termination"))
PSBamp_termination <- phyloseq_to_ampvis2(ps0_termination)  

#Bacteriome

amp_heatmap(PSBamp_termination,
            group_by = "Treatment",
            facet_by = "Sample_time_point",
            plot_values = FALSE,
            tax_show = 30,
            tax_aggregate = "Genus" ,
            #tax_add = "Family",
            tax_empty = "best",
            normalise = TRUE,
            color_vector = c("white", "red"),
            plot_colorscale = "sqrt",
            plot_legendbreaks = c(1, 10, 25, 50)
) +
  theme_classic() +
  # facet_wrap(~Sample_time_point)+
  theme(strip.background = element_rect(colour = "black", fill = "white"),
        strip.text.x = element_text(face = "bold"),
        axis.text.x=element_text(angle = -45, hjust = 0,vjust = 0.2))
```




# Alpha diversity

## By Treatment

```{r, eval = T, include = T, echo = F, message = F, warning=FALSE}

#Normalize to mean read count
standf = function(x, t=total) round(t * (x / sum(x)))
total = median(sample_sums(PSB))
PSB.R = transform_sample_counts(PSB, standf)

#Set desired alpha diversity metrics

# alpha_met <- c("Observed","Shannon","InvSimpson")

alpha_met <- c("Shannon")

###########Boxplots

##Time

variable <- c("Treatment")
# variable <- c("Sample_time_point")
#Set sign argumetns
symnum.args <- list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))
my_comparisons = list( c("CP_CV", "CP_FVT"), c("CP_CV", "LGG_FVT"), c("CP_CV", "AM_FVT"), c("CP_CV", "LGG_CV"), c("CP_CV", "AM_CV"), c("AM_FVT", "AM_CV"), c("LGG_FVT", "LGG_CV") ) #All comparisons


alpha.Treatment <- plot_richness(PSB.R, measures=alpha_met , x=variable, color=variable) + 
  geom_boxplot(alpha=0.1, lwd = 0.5) +
  #stat_compare_means(comparisons = my_comparisons, method = "kruskal", paired = FALSE, symnum.args = symnum.args) +
  scale_color_jco() +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+                   #place different group into sample figure
  theme_classic() +
  theme(text = element_text(size = 8),
        axis.line=element_line(size=1),
        axis.text=element_text(size = 8, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(size = 8, angle = 0),
        legend.text = element_text(size = 8)
  ) +
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )+
  # scale_color_discrete(limits=c("LFD_control","HFD_control","FVT_UnT", "FVT_ChP", "FVT_SDT", "FVT_PyT"))+
  labs(x="Treatment",
       y="Alpha diversity")


alpha.Treatment
```


##Anova and linear regression analysis

```{r, eval = T, include = T, echo = F, warning = F}
##Richness - all time points

richness = estimate_richness(PSB.R, measures = alpha_met)

#Time

rich <- cbind(richness, variable = sample_data(PSB.R)$Treatment)

TukeyHSD(aov(Shannon ~ variable, rich))

rich$variable <- fct_relevel(rich$variable, "LFD_control")
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "HFD_control") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)

rich$variable <- fct_relevel(rich$variable, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


```


## Arrival - By Treatment - Shannon

```{r, eval = T, include = T, echo = F, message = F, warning=FALSE}

PSB_Arrival <- subset_samples(PSB, Sample_time_point %in% c("Arrival"))


#Normalize to mean read count
standf = function(x, t=total) round(t * (x / sum(x)))
total = median(sample_sums(PSB_Arrival))
PSB_Arrival.R = transform_sample_counts(PSB_Arrival, standf)


#Set desired alpha diversity metrics

# alpha_met <- c("Observed","Shannon","InvSimpson")

alpha_met <- c("Shannon")

###########Boxplots

##Time

variable <- c("Treatment")
# variable <- c("Sample_time_point")
#Set sign argumetns
symnum.args <- list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))
# my_comparisons = list( c("CP_CV", "CP_FVT"), c("CP_CV", "LGG_FVT"), c("CP_CV", "AM_FVT"), c("CP_CV", "LGG_CV"), c("CP_CV", "AM_CV"), c("AM_FVT", "AM_CV"), c("LGG_FVT", "LGG_CV") ) #All comparisons


alpha.Treatment <- plot_richness(PSB_Arrival.R, measures=alpha_met , x=variable, color=variable) + 
  geom_boxplot(alpha=0.1, lwd = 1) +
  #stat_compare_means(comparisons = my_comparisons, method = "kruskal", paired = FALSE, symnum.args = symnum.args) +
  scale_color_jco() +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+                   #place different group into sample figure
  theme_classic() +
  theme(text = element_text(size = 8),
        axis.line=element_line(size=1),
        axis.text=element_text(size = 8, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(size = 8, angle = 0),
        legend.text = element_text(size = 8)
  ) +
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )+
  # scale_color_discrete(limits=c("LFD_control","HFD_control","FVT_UnT", "FVT_ChP", "FVT_SDT", "FVT_PyT"))+
  labs(x="Treatment",
       y="Alpha diversity")

alpha.Treatment
```
##Arrival - Anova and linear regression analysis

```{r, eval = T, include = T, echo = F, warning = F}
##Richness - all time points

richness = estimate_richness(PSB_Arrival.R, measures = alpha_met)

#Time

rich <- cbind(richness, variable = sample_data(PSB_Arrival.R)$Treatment)

TukeyHSD(aov(Shannon ~ variable, rich))

rich$variable <- fct_relevel(rich$variable, "LFD_control")
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "HFD_control") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


```

## Before_1st_FVT - By Treatment - Shannon

```{r, eval = T, include = T, echo = F, message = F, warning=FALSE}

PSB_Before_1st_FVT <- subset_samples(PSB, Sample_time_point %in% c("Before_1st_FVT"))


#Normalize to mean read count
standf = function(x, t=total) round(t * (x / sum(x)))
total = median(sample_sums(PSB_Before_1st_FVT))
PSB_Before_1st_FVT.R = transform_sample_counts(PSB_Before_1st_FVT, standf)


#Set desired alpha diversity metrics

# alpha_met <- c("Observed","Shannon","InvSimpson")

alpha_met <- c("Shannon")

###########Boxplots

##Time

variable <- c("Treatment")
# variable <- c("Sample_time_point")
#Set sign argumetns
symnum.args <- list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))
# my_comparisons = list( c("CP_CV", "CP_FVT"), c("CP_CV", "LGG_FVT"), c("CP_CV", "AM_FVT"), c("CP_CV", "LGG_CV"), c("CP_CV", "AM_CV"), c("AM_FVT", "AM_CV"), c("LGG_FVT", "LGG_CV") ) #All comparisons


alpha.Treatment <- plot_richness(PSB_Before_1st_FVT.R, measures=alpha_met , x=variable, color=variable) + 
  geom_boxplot(alpha=0.1, lwd = 1) +
  #stat_compare_means(comparisons = my_comparisons, method = "kruskal", paired = FALSE, symnum.args = symnum.args) +
  scale_color_jco() +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+                   #place different group into sample figure
  theme_classic() +
  theme(text = element_text(size = 8),
        axis.line=element_line(size=1),
        axis.text=element_text(size = 8, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(size = 8, angle = 0),
        legend.text = element_text(size = 8)
  ) +
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )+
  # scale_color_discrete(limits=c("LFD_control","HFD_control","FVT_UnT", "FVT_ChP", "FVT_SDT", "FVT_PyT"))+
  labs(x="Treatment",
       y="Alpha diversity")

alpha.Treatment
```
## Before_1st_FVT  - Anova and linear regression analysis

```{r, eval = T, include = T, echo = F, warning = F}
##Richness - all time points

richness = estimate_richness(PSB_Before_1st_FVT.R, measures = alpha_met)

#Time

rich <- cbind(richness, variable = sample_data(PSB_Before_1st_FVT.R)$Treatment)

TukeyHSD(aov(Shannon ~ variable, rich))

rich$variable <- fct_relevel(rich$variable, "LFD_control")
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "HFD_control") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


```

## 1w_after_FVT - By Treatment - Shannon

```{r, eval = T, include = T, echo = F, message = F, warning=FALSE}

PSB_1w_after_FVT <- subset_samples(PSB, Sample_time_point %in% c("1w_after_FVT"))


#Normalize to mean read count
standf = function(x, t=total) round(t * (x / sum(x)))
total = median(sample_sums(PSB_1w_after_FVT))
PSB_1w_after_FVT.R = transform_sample_counts(PSB_1w_after_FVT, standf)


#Set desired alpha diversity metrics

# alpha_met <- c("Observed","Shannon","InvSimpson")

alpha_met <- c("Shannon")

###########Boxplots

##Time

variable <- c("Treatment")
# variable <- c("Sample_time_point")
#Set sign argumetns
symnum.args <- list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))
# my_comparisons = list( c("CP_CV", "CP_FVT"), c("CP_CV", "LGG_FVT"), c("CP_CV", "AM_FVT"), c("CP_CV", "LGG_CV"), c("CP_CV", "AM_CV"), c("AM_FVT", "AM_CV"), c("LGG_FVT", "LGG_CV") ) #All comparisons


alpha.Treatment <- plot_richness(PSB_1w_after_FVT.R, measures=alpha_met , x=variable, color=variable) + 
  geom_boxplot(alpha=0.1, lwd = 1) +
  #stat_compare_means(comparisons = my_comparisons, method = "kruskal", paired = FALSE, symnum.args = symnum.args) +
  scale_color_jco() +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+                   #place different group into sample figure
  theme_classic() +
  theme(text = element_text(size = 8),
        axis.line=element_line(size=1),
        axis.text=element_text(size = 8, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(size = 8, angle = 0),
        legend.text = element_text(size = 8)
  ) +
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )+
  # scale_color_discrete(limits=c("LFD_control","HFD_control","FVT_UnT", "FVT_ChP", "FVT_SDT", "FVT_PyT"))+
  labs(x="Treatment",
       y="Alpha diversity")

alpha.Treatment
```
## 1w_after_FVT  - Anova and linear regression analysis

```{r, eval = T, include = T, echo = F, warning = F}
##Richness - all time points

richness = estimate_richness(PSB_1w_after_FVT.R, measures = alpha_met)

#Time

rich <- cbind(richness, variable = sample_data(PSB_1w_after_FVT.R)$Treatment)

TukeyHSD(aov(Shannon ~ variable, rich))

rich$variable <- fct_relevel(rich$variable, "LFD_control")
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "HFD_control") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


```
## Termination - By Treatment - Shannon

```{r, eval = T, include = T, echo = F, message = F, warning=FALSE}

PSB_Termination <- subset_samples(PSB, Sample_time_point %in% c("Termination"))


#Normalize to mean read count
standf = function(x, t=total) round(t * (x / sum(x)))
total = median(sample_sums(PSB_Termination))
PSB_Termination.R = transform_sample_counts(PSB_Termination, standf)


#Set desired alpha diversity metrics

# alpha_met <- c("Observed","Shannon","InvSimpson")

alpha_met <- c("Shannon")

###########Boxplots

##Time

variable <- c("Treatment")
# variable <- c("Sample_time_point")
#Set sign argumetns
symnum.args <- list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))
# my_comparisons = list( c("CP_CV", "CP_FVT"), c("CP_CV", "LGG_FVT"), c("CP_CV", "AM_FVT"), c("CP_CV", "LGG_CV"), c("CP_CV", "AM_CV"), c("AM_FVT", "AM_CV"), c("LGG_FVT", "LGG_CV") ) #All comparisons


alpha.Treatment <- plot_richness(PSB_Termination.R, measures=alpha_met , x=variable, color=variable) + 
  geom_boxplot(alpha=0.1, lwd = 1) +
  #stat_compare_means(comparisons = my_comparisons, method = "kruskal", paired = FALSE, symnum.args = symnum.args) +
  scale_color_jco() +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+                   #place different group into sample figure
  theme_classic() +
  theme(text = element_text(size = 8),
        axis.line=element_line(size=1),
        axis.text=element_text(size = 8, colour = "Black"),
        axis.ticks=element_line(size=1, colour = "Black"),
        strip.background = element_rect(colour = "white", fill = "white"),
        axis.text.x=element_text(angle = 45, hjust = 1),
        strip.text.x = element_text(size = 8, angle = 0),
        legend.text = element_text(size = 8)
  ) +
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )+
  # scale_color_discrete(limits=c("LFD_control","HFD_control","FVT_UnT", "FVT_ChP", "FVT_SDT", "FVT_PyT"))+
  labs(x="Treatment",
       y="Alpha diversity")

alpha.Treatment
```
## Termination  - Anova and linear regression analysis

```{r, eval = T, include = T, echo = F, warning = F}
##Richness - all time points

richness = estimate_richness(PSB_Termination.R, measures = alpha_met)

#Time

rich <- cbind(richness, variable = sample_data(PSB_Termination.R)$Treatment)

TukeyHSD(aov(Shannon ~ variable, rich))

rich$variable <- fct_relevel(rich$variable, "LFD_control")
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "HFD_control") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


rich$variable <- fct_relevel(rich$variable, "FVT_UnT") #Releveling to control 
fit_norm <- lm(Shannon ~variable, data = rich)
summary(fit_norm)


```

# Beta diversity

## Abbreviation - basically the mouse ID. 


```{r, eval = T, include = T, echo = F, massage = F}
#Do PCoA ordination based on Bray-curtis distance
GP.ord <- ordinate(PSB.CSS, "PCoA", "bray")

#####################Run + Batch

#PSB.CSS@sam_data$Run <- as.factor(PSB.CSS@sam_data$Run)

beta.id = phyloseq::plot_ordination(PSB.CSS, GP.ord, color="Treatment") + 
  #stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  #scale_fill_manual(values= col_fil) +
  #scale_color_manual(values = col_fil) +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank()
  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
  geom_dl(aes(label = Treatment), method = list(dl.trans(x = x + 0.0), "last.points", cex = 0.8))
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.id
```

## Treatment??axis1??2

```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, axes = 1:2,color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment
```

## Treatment??axis1??3

```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, axes= 1:3, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment
```

## Treatment??axis 2??3

```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, axes= 2:3, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment
```

## Treatment??axis 2??3 3d
```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, axes= 2:3, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)
      
  ) + 
  geom_point()+
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment

```


## Treatment??axis 3??4

```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, axes= 3:4, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment
```

## Treatment??axis 1??4

```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, axes= 1:4, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment
```

## Treatment??axis 1??4 Arrival

```{r, eval = T, include = T, echo = F, message = F}
PSB_Arrival <- subset_samples(PSB.CSS, Sample_time_point %in% c("Arrival"))

beta.treatment = phyloseq::plot_ordination(PSB_Arrival, GP.ord, axes= 1:4, color="Treatment") +
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment
```



## Treatment??axis 2??4

```{r, eval = T, include = T, echo = F, message = F}
beta.treatment = phyloseq::plot_ordination(PSB.CSS, GP.ord, axes= 2:4, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 1) +
  facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment
```
## Adonis and pairwise permanova analysis - Treatment
```{r, eval = T, include = T, echo = F, message = F}
library(mctoolsr)

bray.PSB <- phyloseq::distance(PSB.CSS, method = "bray")
# make a data frame from the sample_data
sampledf.PSB <- data.frame(sample_data(PSB.CSS))

adonis.PSB <- adonis2(bray.PSB ~ Treatment, method = "bray", data = sampledf.PSB, permutations = 999)
adonis.PSB

mapping_file = data.frame(sample_data(PSB.CSS))
pairwiseperm <- mctoolsr::calc_pairwise_permanovas(bray.PSB, mapping_file, "Treatment")
pairwiseperm
```


## Arrival - Treatment

```{r, eval = T, include = T, echo = F, message = F}
PSB.CSS_Arrival <- subset_samples(PSB.CSS, Sample_time_point %in% c("Arrival"))

GP.ord_Arrival <- ordinate(PSB.CSS_Arrival, "PCoA", "bray")


beta.treatment1 = phyloseq::plot_ordination(PSB.CSS_Arrival, GP.ord_Arrival, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 0.5) +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 10, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  # ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment1
```

## Arrival - Adonis and pairwise permanova analysis - Treatment
```{r, eval = T, include = T, echo = F, message = F}
library(mctoolsr)

bray.PSB <- phyloseq::distance(PSB.CSS_Arrival, method = "bray")
# make a data frame from the sample_data
sampledf.PSB <- data.frame(sample_data(PSB.CSS_Arrival))

adonis.PSB <- adonis2(bray.PSB ~ Treatment, method = "bray", data = sampledf.PSB, permutations = 999)
adonis.PSB

mapping_file = data.frame(sample_data(PSB.CSS_Arrival))
pairwiseperm <- mctoolsr::calc_pairwise_permanovas(bray.PSB, mapping_file, "Treatment")
pairwiseperm
```

## Before_1st_FVT - Treatment

```{r, eval = T, include = T, echo = F, message = F}
PSB.CSS_Before_1st_FVT <- subset_samples(PSB.CSS, Sample_time_point %in% c("Before_1st_FVT"))

GP.ord_Before_1st_FVT <- ordinate(PSB.CSS_Before_1st_FVT, "PCoA", "bray")


beta.treatment2 = phyloseq::plot_ordination(PSB.CSS_Before_1st_FVT, GP.ord_Before_1st_FVT, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 0.5) +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 10, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  # ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment2
```

## Before_1st_FVT - Adonis and pairwise permanova analysis - Treatment
```{r, eval = T, include = T, echo = F, message = F}
library(mctoolsr)

bray.PSB <- phyloseq::distance(PSB.CSS_Before_1st_FVT, method = "bray")
# make a data frame from the sample_data
sampledf.PSB <- data.frame(sample_data(PSB.CSS_Before_1st_FVT))

adonis.PSB <- adonis2(bray.PSB ~ Treatment, method = "bray", data = sampledf.PSB, permutations = 999)
adonis.PSB

mapping_file = data.frame(sample_data(PSB.CSS_Before_1st_FVT))
pairwiseperm <- mctoolsr::calc_pairwise_permanovas(bray.PSB, mapping_file, "Treatment")
pairwiseperm
```
## 1w_after_FVT - Treatment

```{r, eval = T, include = T, echo = F, message = F}
PSB.CSS_1w_after_FVT <- subset_samples(PSB.CSS, Sample_time_point %in% c("1w_after_FVT"))

GP.ord_1w_after_FVT <- ordinate(PSB.CSS_1w_after_FVT, "PCoA", "bray")


beta.treatment3 = phyloseq::plot_ordination(PSB.CSS_1w_after_FVT, GP.ord_1w_after_FVT, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 0.5) +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 10, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
                axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)

  ) + 
  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  # ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment3
```

## 1w_after_FVT - Adonis and pairwise permanova analysis - Treatment
```{r, eval = T, include = T, echo = F, message = F}
library(mctoolsr)

bray.PSB <- phyloseq::distance(PSB.CSS_1w_after_FVT, method = "bray")
# make a data frame from the sample_data
sampledf.PSB <- data.frame(sample_data(PSB.CSS_1w_after_FVT))

adonis.PSB <- adonis2(bray.PSB ~ Treatment, method = "bray", data = sampledf.PSB, permutations = 999)
adonis.PSB

mapping_file = data.frame(sample_data(PSB.CSS_1w_after_FVT))
pairwiseperm <- mctoolsr::calc_pairwise_permanovas(bray.PSB, mapping_file, "Treatment")
pairwiseperm
```
## Termination - Treatment

```{r, eval = T, include = T, echo = F, message = F}
PSB.CSS_Termination <- subset_samples(PSB.CSS, Sample_time_point %in% c("Termination"))

GP.ord_Termination <- ordinate(PSB.CSS_Termination, "PCoA", "bray")


beta.treatment4 = phyloseq::plot_ordination(PSB.CSS_Termination, GP.ord_Termination, color="Treatment") + 
  stat_ellipse(geom = "polygon", level = 0.95, fill = NA, size = 0.5) +
  # facet_wrap(~Sample_time_point, nrow = 1, ncol = 4)+  
  scale_color_jco() +
  theme_classic() + 
  theme(plot.title = element_text(size = 10, face = "bold",hjust = 0.5),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 10),
        axis.text.y = element_text(size = 10)
  ) + 

  #geom_point(aes(shape = C.section..Yes.No.), size = 3) +
  scale_x_reverse() +# Revert x-axis to match plots
  # ggtitle("Bacteriome")+
                  scale_color_manual(
                    values=c("#F27970","#BB9727","#54B345","#32B897","#05B9E2","#8983BF"),
                    breaks = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control"),
                    labels = c( "FVT_ChP", "FVT_SDT", "FVT_PyT","FVT_UnT","HFD_control","LFD_control")
                  )
#ggplot2::annotate("text", x = -0.3, y = 0.3, label = "P < 0.001") #Add p-value from adonis

beta.treatment4
```

## Termination - Adonis and pairwise permanova analysis - Treatment
```{r, eval = T, include = T, echo = F, message = F}
library(mctoolsr)

bray.PSB <- phyloseq::distance(PSB.CSS_Termination, method = "bray")
# make a data frame from the sample_data
sampledf.PSB <- data.frame(sample_data(PSB.CSS_Termination))

adonis.PSB <- adonis2(bray.PSB ~ Treatment, method = "bray", data = sampledf.PSB, permutations = 999)
adonis.PSB

mapping_file = data.frame(sample_data(PSB.CSS_Termination))
pairwiseperm <- mctoolsr::calc_pairwise_permanovas(bray.PSB, mapping_file, "Treatment")
pairwiseperm
```


## Combine 4 time point
```{r}
Figure.beta.4timepoint <- ggarrange(beta.treatment1,beta.treatment2, beta.treatment3, beta.treatment4,
                  labels = c("Arrival","Before 1st FVT", "1w after 2nd FVT", "Termination"),
                  ncol = 4, nrow = 1,
                  common.legend = TRUE,
                  font.label = list(size = 12))+
                  ggtitle("Bacteriome")

Figure.beta.4timepoint
```

## Combine 3 time point
```{r}
Figure.beta.3timepoint <- ggarrange(beta.treatment1, beta.treatment3, beta.treatment4,
                  labels = c("Arrival", "1w after 2nd FVT", "Termination"),
                  ncol = 3, nrow = 1,
                  common.legend = TRUE,
                  font.label = list(size = 12))+
                  ggtitle("Bacteriome")

Figure.beta.3timepoint
```


### Bacteriome - Deseq2 - Treatment - Collapsed on Genus-level - Comparing ChP vs HFD. 
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- tax_glom(PSB, "Genus", NArm = FALSE)
#NB!! Here you choose which groups to compare in the DeSeq2 - Try the different ones yourself
PSB.pseu <- subset_samples(PSB, Treatment %in% c("FVT_ChP","HFD_control"))

otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)



diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds = DESeq2::DESeq(diagdds, test="Wald", fitType="parametric")

res = results(diagdds, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  #theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Treatment")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel))


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)

#Remove Realm from tax_table
#PSB.diff.rel@tax_table <- tax_table(PSB.diff.rel)[,c(1,3:8)]

ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic() +
  theme(axis.text.x=element_text(angle = 45, hjust = 0.9,vjust = 0.95)) +
  ggtitle("Differentially abundant bacterial species")
```


### Bacteriome - Deseq2 - Treatment - Collapsed on Genus-level - here the software itself choose a random "control" to compare with. 

```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- tax_glom(PSB, "Genus", NArm = FALSE)

PSB.pseu <- PSB.pseu

#PSB.pseu@sam_data$Days <- as.factor(PSB.pseu@sam_data$Days)

otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)



diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds = DESeq2::DESeq(diagdds, test="Wald", fitType="parametric")

res = results(diagdds, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  #theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Treatment")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.rel, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel))


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)

#Remove Realm from tax_table
#PSB.diff.rel@tax_table <- tax_table(PSB.diff.rel)[,c(1,3:8)]

ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Genus",
            tax_show = 20,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic() +
  theme(axis.text.x=element_text(angle = 45, hjust = 0.9,vjust = 0.95)) +
  ggtitle("Differentially abundant bacterial species")
```

### Bacteriome - Deseq2 - Treatment

```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- PSB

#PSB.pseu@sam_data$Days <- as.factor(PSB.pseu@sam_data$Days)

otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds = DESeq2::DESeq(diagdds, test="Wald", fitType="parametric")

res = results(diagdds, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  #theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Treatment")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB.pseu, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel))


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)

#Remove Realm from tax_table
#PSB.diff.rel@tax_table <- tax_table(PSB.diff.rel)[,c(1,3:8)]

ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Species",
            tax_show = 15,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic() +
  theme(axis.text.x=element_text(angle = 45, hjust = 0.9,vjust = 0.95)) +
  ggtitle("Differentially abundant bacterial species")
```

#Deseq2 - defined comparison
```{r, eval = T, include = T, echo = F, message = F, warning=F}
#Settings for cutoff values for plotting
alpha = 0.05 #minimum p-value
beta = 2 #minimum log-fold difference 

PSB.pseu <- subset_samples(PSB, Treatment %in% c("FVT_UnT","HFD_control"))


otu_table(PSB.pseu) = otu_table(PSB.pseu) + 1 #Add spudocount of one to avoid DESeq2 error(log transformation fill be performed anyways)

diagdds = phyloseq_to_deseq2(PSB.pseu, ~ Treatment)
diagdds.nw_ob = DESeq2::DESeq(diagdds, test="Wald", fitType="mean", sfType = "ratio")

res = results(diagdds.nw_ob, cooksCutoff = FALSE)

sigtab = res[which(res$padj < alpha & abs(res$log2FoldChange) > beta),] #Select values from results file with adj p values>alpha AND log2fold cahnge>beta
sigtab = cbind(as(sigtab, "data.frame"), as.matrix(tax_table(PSB)[rownames(sigtab), ]))
#head(sigtab)

#Plotting - genus
# Order order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Order = factor(as.character(sigtab$Order), levels=names(x))
# Genus order based on abundance 
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

diffplot <- ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Order)) + geom_point(size=6) + 
  scale_colour_jco() +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))+
  ggtitle("Differential abundance by Gestational age")


##Heatmap of differentially abundant species

#PSB.diff <- merge_phyloseq(subset(otu_table(PSB), rownames(otu_table(PSB)) %in% rownames(sigtab)),tax_table(PSB), sample_data(PSB))

PSB.rel <- microbiome::transform(PSB, "compositional")

PSB.diff.rel <- merge_phyloseq(subset(otu_table(PSB.rel), rownames(otu_table(PSB.rel)) %in% rownames(sigtab)),tax_table(PSB.rel), sample_data(PSB.rel)) 


#glom <- tax_glom(subset_taxa(PSB.diff, Kingdom != ""), "Species", NArm = FALSE)


ampvis2_PSB.diff.rel <- phyloseq_to_ampvis2(PSB.diff.rel) #Convert phyloseq to ampvis2 object

ampvis2_PSB.diff.rel$abund <- ampvis2_PSB.diff.rel$abund*100 #Multiply relative abundance by 100 for read percentage

amp_heatmap(ampvis2_PSB.diff.rel,
            group_by = "Treatment",
            tax_aggregate = "Species",
            tax_show = 15,
            normalise = FALSE,
            tax_empty = "best",
            tax_add = "Family") +
  theme_classic()

```

## Bacteriome - Effect-size

```{r, eval = T, include = F, echo = F, message = F}

#################################Importing-Data

#standf = function(x, t=total) round(t * (x / sum(x)))
#total = median(sample_sums(PSB))
#PSB.R = transform_sample_counts(PSB, standf)

import.otu.table <- as.data.frame(otu_table(PSB.CSS)) %>%
  #dplyr::select(-taxonomy) %>%
  mutate_all(as.numeric) #Convert to numeric where possible



cure.data = meta.bac

#import.taxonomy = read.table('Bacteriome_taxonomy_LCA_known.txt', header = T)

rownames <- rownames(cure.data)

#Remove NA colomns

cure.data <- cure.data[ , ! apply( cure.data , 2 , function(x) all(is.na(x)) ) ]

#Set numeric column class
cure.data <- cure.data %>% mutate_if(is.character,as.factor)
#Add back rownames

rownames(cure.data) <- rownames

#cure.data <- cure.data %>% mutate_if(is.character, as.numeric)

X8.nonrarefied = t(import.otu.table)                                                                  ###%%% non-rarefied


###rarefying|subsampling reads in every sample
barplot(colSums(import.otu.table))
min(colSums(import.otu.table))

rare8 <- t(import.otu.table)
#rare8 = rarefy(t(import.otu.table), sample = 450000)
X8.rarefied = as.data.frame(rare8)                                                                 ###%%% rarefied

###Subset metadata to match remaining samples after rarefaction

Y.all = cure.data[rownames(X8.rarefied),]

###Log tranformation
X = log10(X8.rarefied+1)

#Select relevant numeric column

#Y <- dplyr::select(Y.all,c("Diet", "Treatment", "AUC1_week18", "AUC2_week18", "ewat_mg", "Weight_gain_perc" , "Central_memory_T_cells_CD4_Fat", "Effector_memory_T_cells_CD4_Fat", "Naive_T_cells_CD4_Fat", "Central_memory_T_cells_CD8_Fat", "Effector_memory_T_cells_CD8_Fat", "Naive_T_cells_CD8_Fat", "CD8a_TCRab_Fat","B_cells_Fat", "Dendritic_cells_Fat","Macrophages_Fat", "M1_macrophages_Fat", "GM.CSF", "IFN.g", "IL.10", "IL.15", "IL.17_A_F", "IL.6", "KC_GRO", "MIP.2","TNF.a", "IL.22"))

#Here you need to adjust to what is found in the mapping file. So if one timepoint does not have any data, the columns shold not be included. 
Y <- dplyr::select(Y.all,c("Diet", "Treatment","ewat_mg", "Weight_gain_perc","AUC1_week18", "AUC2_week18","Central_memory_T_cells_CD8_Fat", "Th_cells_Fat","M1_macrophages_Fat","B_cells_Fat","Dendritic_cells_Fat", "MIP.2","TNF.a","IL.15","IL.10"))

#Y <- dplyr::select(Y.all,c("Diet", "Treatment", "AUC1_week18", "AUC2_week18", "ewat_mg", "Weight_gain_perc" , "Central_memory_T_cells_CD4_Fat", "Effector_memory_T_cells_CD4_Fat", "Naive_T_cells_CD4_Fat", "Central_memory_T_cells_CD8_Fat", "Effector_memory_T_cells_CD8_Fat", "Naive_T_cells_CD8_Fat", "CD8a_TCRab_Fat","B_cells_Fat", "Dendritic_cells_Fat","Macrophages_Fat", "M1_macrophages_Fat", "GM.CSF", "IFN.g", "IL.10", "IL.15", "IL.17_A_F", "IL.6", "KC_GRO", "MIP.2","TNF.a", "IL.22"))

Y <- unfactor(Y)
Y <- dplyr::mutate_at(Y,1:2,funs(factor)) 

#Y <- dplyr::mutate_if(Y,is.factor, as.numeric)
#Y <- dplyr::mutate_at(Y,1:5,funs(factor)) 
#Y <- dplyr::select(Y.all,c("Bact_group", "FVT_group", "Treatment", "FVT_Treatment", "Gender"))


colnames(Y) #Check remaining columns

###Select only numeric metadata culumns

#Y  <- Y %>% mutate_all(as.numeric)

##vifcor
#collinearity <- vifcor(Y, th  =0.5) #at 0.5 threshold variables have a VIF below 10.00 (<0.21 correlation!)
#collinearity

#Select relevant columns for that are non-collinear

Y.ad = Y
#Y.ad <- dplyr::select(Y.all,c("Batch", "Time")

#Y.ad <- Y.all[8:23]

#Remove NA rows

Y.ad <- Y.ad %>% drop_na()

X.ad <- X[rownames(Y.ad),]

is.na(Y.ad)
```

## Non-constrained - Consider du run this without the LFD group - since there obviosuly are quite some difference.
#### Capscale - independent effect sizes

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE, fig.align="center"}
###############Anova on capscale - all variables

anova.m <- matrix(nrow = 0, ncol =6) #Create matrix for results
colnames(anova.m) <- c("Df","SumOfSqs","F","Pr.F","Residual.Df","Residual.SumOfSqs")                
for (i in colnames(Y)){
  anova <- as.matrix(anova(capscale(X ~ Y[,i], dist = "bray")))
  anova <- c(anova[1,],anova[2,1:2])
  anova.m <-rbind(anova.m,anova)
}
rownames(anova.m) <- colnames(Y)

#Prepare for plotting
anova.dat <- anova.m %>%
  as.data.frame %>%
  tibble::rownames_to_column("Variable") %>%
  arrange(F)

#Add adjusted p-values
anova.dat$Pr.F.adj <- p.adjust(anova.dat$Pr.F, method = 'fdr', n = ncol (Y.ad))

#Show table
#knitr::kable(anova.dat)

#Divide F by 10
anova.dat$F <- anova.dat$F / 10

##Fancy plot

anova.melt <- tidyr::pivot_longer(anova.dat,cols=c("F"), names_to='measure', values_to="value") %>%
  add_significance(p.col = "Pr.F.adj", output.col = "PrF.sym",
                   cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1),
                   symbols = c("***", "**", "*", "'", "")
                   )




fancy.anova.bar <- ggbarplot(anova.melt,
                                         x="Variable",
                                         y="value",
                                         #color = "Variable",
                                         orientation = c("horizontal"),
                                         color = NA,
                                         fill = "measure",
                                         position=position_dodge(0.7),
                                         label = anova.melt$PrF.sym,
                                         lab.size = 9,
                                         lab.vjust = 0.7,
                                         lab.hjust = -0.3
                                         
) +
  #stat_pvalue_manual(
  #  adonis.melt,  label = "PrF.sym", tip.length = 0.01
  #) +
  #geom_hline(yintercept = 0.05) +
  ggtitle("Capscale dbRDA effect size: all variables") +
  ylab(expression(R^2~"value")) +
  scale_fill_jco()

fancy.anova.bar

```

#### ADONIS - decomposed to show individual contributions of factors

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE, fig.align="center"}
##All variables

adonis <- adonis(X ~ ., data = Y, method = "bray") #All variables

#Add adjusted p-values
adonis$aov.tab$`Pr(>F).adj` <- p.adjust(adonis$aov.tab$`Pr(>F)`, method = 'fdr', n = ncol (Y.ad))

#Show table
#knitr::kable(anova.dat)

#Plotting ordered R2
adonis.dat <- as.data.frame(adonis$aov.tab) %>%
  tibble::rownames_to_column("Variable")  %>%
  filter(!Variable %in% c("Residuals","Total")) %>%
  arrange(R2)

adonis.melt <- tidyr::pivot_longer(adonis.dat,cols=c("R2"), names_to='measure', values_to="value") %>%
  add_significance(p.col = "Pr(>F).adj", output.col = "PrF.sym",
                   cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1),
                   symbols = c("***", "**", "*", "'", "")
                   )



fancy.adonis.bar <- ggbarplot(adonis.melt,
                                         x="Variable",
                                         y="value",
                                         #color = "Variable",
                                         orientation = c("horizontal"),
                                         color = NA,
                                         fill = "measure",
                                         position=position_dodge(0.7),
                                         label = adonis.melt$PrF.sym,
                                         lab.size = 9,
                                         lab.vjust = 0.7,
                                         lab.hjust = -0.3
                                         
) +
  #stat_pvalue_manual(
  #  adonis.melt,  label = "PrF.sym", tip.length = 0.01
  #) +
  #geom_hline(yintercept = 0.05) +
  ggtitle("Adonis effect size: non-collinear variables") +
  ylab(expression(R^2~"value")) +
  scale_fill_jco()

fancy.adonis.bar
```

### Constrained by Diet - Here you can remove factors from the equation. E.g. diet  and thereby also the effect of LFD
#### Capscale - independent effect sizes

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE, fig.align="center"}
###############Anova on capscale -Constrained for diet

anova.m <- matrix(nrow = 0, ncol =6) #Create matrix for results
colnames(anova.m) <- c("Df","SumOfSqs","F","Pr.F","Residual.Df","Residual.SumOfSqs")                
for (i in colnames(dplyr::select(Y,-c("Diet")))){
  anova <- as.matrix(anova(capscale(X ~ Y[,i] + Condition(Diet), Y, dist = "bray")), Y, dist = "bray")
  anova <- c(anova[1,],anova[2,1:2])
  anova.m <-rbind(anova.m,anova)
}
rownames(anova.m) <- colnames(dplyr::select(Y,-c("Diet")))

#Prepare for plotting
anova.dat <- anova.m %>%
  as.data.frame %>%
  tibble::rownames_to_column("Variable") %>%
  arrange(F)

#Add adjusted p-values
anova.dat$Pr.F.adj <- p.adjust(anova.dat$Pr.F, method = 'fdr', n = ncol (Y.ad))

#Show table
#knitr::kable(anova.dat)

#Divide F by 10
anova.dat$F <- anova.dat$F / 10

##Fancy plot

anova.melt <- tidyr::pivot_longer(anova.dat,cols=c("F"), names_to='measure', values_to="value") %>%
  add_significance(p.col = "Pr.F.adj", output.col = "PrF.sym",
                   cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1),
                   symbols = c("***", "**", "*", "'", "")
                   )

fancy.anova.bar <- ggbarplot(anova.melt,
                                         x="Variable",
                                         y="value",
                                         #color = "Variable",
                                         orientation = c("horizontal"),
                                         color = NA,
                                         fill = "measure",
                                         position=position_dodge(0.7),
                                         label = anova.melt$PrF.sym,
                                         lab.size = 9,
                                         lab.vjust = 0.7,
                                         lab.hjust = -0.3
                                         
) +
  #stat_pvalue_manual(
  #  adonis.melt,  label = "PrF.sym", tip.length = 0.01
  #) +
  #geom_hline(yintercept = 0.05) +
  ggtitle("Capscale dbRDA effect size: constrained for Diet") +
  ylab(expression(R^2~"value")) +
  scale_fill_jco()

fancy.anova.bar

```

#### ADONIS - decomposed to show individual contributions of factors. Contrained for Diet

```{r, eval = T, include = T, echo = F, message = F, warning =  F, fig.width=10, fig.height=4, fig.fullwidth = TRUE, fig.align="center"}
##All variables

adonis <- adonis(X ~ ., data = Y, method = "bray", strata = Y$Diet)

#Add adjusted p-values
adonis$aov.tab$`Pr(>F).adj` <- p.adjust(adonis$aov.tab$`Pr(>F)`, method = 'fdr', n = ncol (Y))

#Show table
#knitr::kable(anova.dat)

#Plotting ordered R2
adonis.dat <- as.data.frame(adonis$aov.tab) %>%
  tibble::rownames_to_column("Variable")  %>%
  filter(!Variable %in% c("Residuals","Total")) %>%
  arrange(R2)

adonis.melt <- tidyr::pivot_longer(adonis.dat,cols=c("R2"), names_to='measure', values_to="value") %>%
  add_significance(p.col = "Pr(>F).adj", output.col = "PrF.sym",
                   cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1),
                   symbols = c("***", "**", "*", "'", "")
                   )



fancy.adonis.bar <- ggbarplot(adonis.melt,
                                         x="Variable",
                                         y="value",
                                         #color = "Variable",
                                         orientation = c("horizontal"),
                                         color = NA,
                                         fill = "measure",
                                         position=position_dodge(0.7),
                                         label = adonis.melt$PrF.sym,
                                         lab.size = 9,
                                         lab.vjust = 0.7,
                                         lab.hjust = -0.3
) +
  #stat_pvalue_manual(
  #  adonis.melt,  label = "PrF.sym", tip.length = 0.01
  #) +
  #geom_hline(yintercept = 0.05) +
  ggtitle("Adonis effect size: non-collinear variables - Diet constrained") +
  ylab(expression(R^2~"value")) +
  scale_fill_jco()

fancy.adonis.bar
```



```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```

```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```
```{r, eval = T, include = T, echo = F, message = F}

```



